% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/select_genes.R
\name{select_genes}
\alias{select_genes}
\title{Selects informative genes based on k-nearest neighbour analysis.}
\usage{
select_genes(
  data = NULL,
  distance_method = c("pearson", "cosine", "euclidean", "spearman", "kendall", "binary"),
  noise_level = 5e-05,
  k = 80,
  row_sum = 1,
  fdr = 0.005,
  which_slot = c("data", "sct", "counts"),
  no_dknn_filter = FALSE,
  no_anti_cor = FALSE,
  final_slot = c("data", "sct", "counts"),
  seed = 123
)
}
\arguments{
\item{data}{A matrix, data.frame or Seurat object.}

\item{distance_method}{a character string indicating the method for computing distances (one of "pearson", "cosine", 
"euclidean", spearman or "kendall").}

\item{noise_level}{This parameter controls the fraction of genes with high dknn (ie. noise) whose neighborhood (i.e associated distances) 
will be used to compute simulated DKNN values. A value of 0 means to use all the genes. A value close to 1 means  to use only gene 
with high dknn (i.e close to noise).}

\item{k}{An integer specifying the size of the neighborhood.}

\item{row_sum}{A feature/gene whose row sum is below this threshold will be discarded. Use -Inf to keep all genes.}

\item{fdr}{A numeric value indicating the false discovery rate threshold (range: 0 to 100).}

\item{which_slot}{a character string indicating which slot to use from the input scRNA-seq object (one of "data", "sct" or "counts") to compute distances.}

\item{no_dknn_filter}{a logical indicating whether to skip the k-nearest-neighbors (KNN) filter. If FALSE, all genes are kept for the next steps.}

\item{no_anti_cor}{If TRUE, correlation below 0 are set to zero ("pearson", "cosine", "spearman" "kendall"). This may increase the 
relative weight of positive correlation (as true anti-correlation may be rare).}

\item{final_slot}{This is the slot to export in the final object. In case distance is set to binary, the function will force the usage of 'counts' for distance
computation. However, the final object may contain normalized data (i.e log transformed) extract from the 'data' slot.}

\item{seed}{An integer specifying the random seed to use.}
}
\value{
a ClusterSet class object
}
\description{
This function selects genes based on k-nearest neighbour analysis.
The function takes a seurat object or gene expression matrix as input 
and compute distance to k-nearest neighbour for each gene/feature.
A threshold is set based on permutation analysis and FDR computation.
}
\examples{

# Restrict vebosity to info messages only.
set_verbosity(1)

# Load a dataset
load_example_dataset("7871581/files/pbmc3k_medium")

# Select informative genes
res <- select_genes(pbmc3k_medium,
                    distance = "pearson",
                    row_sum=5)

# Result is a ClusterSet object
is(res)
slotNames(res)

# The selected genes
nrow(res)
head(row_names(res))

}
\references{
- Lopez F.,Textoris J., Bergon A., Didier G., Remy E., Granjeaud
S., Imbert J. , Nguyen C. and Puthier D. TranscriptomeBrowser: a powerful
and flexible toolbox to explore productively the transcriptional landscape
of the Gene Expression Omnibus database. PLoSONE, 2008;3(12):e4001.
}
\author{
Julie Bavais, Sebastien Nin, Lionel Spinelli and Denis Puthier
}
