
<!-- UPPERCASE variable are text to be replaced (see st_report() function). -->
<!-- NB: The code contains several global variables derived from gm_report() function arguments. -->



```{css echo=FALSE}
.panelset .panel-active {
    display: block !important;
}
```

```{r include = FALSE}
# from https://tinyurl.com/mwxj7544
DT::datatable(NULL)
```

```{r start_MOD_MODULE_NUMBER, include=FALSE, echo=FALSE, eval=TRUE}
suppressWarnings(rm(enrich_res))
suppressWarnings(rm(edo))
suppressWarnings(rm(enrich_res_test))
scigenex::set_verbosity(0)
library(knitr)
options(width=400)

knitr::opts_chunk$set(
  fig.path="FIG_PATH",
  fig.align = "center", 
  size = "tiny", 
  comment = "",
  fig.dpi=300,
  out.width="100%"
)
```


```{r, level_0_MOD_MODULE_NUMBER, eval=TRUE, echo=FALSE, results = "asis"} 
knitr::asis_output("# Module MODULE_NUMBER ")
``` 

```{r, level_0b_MOD_MODULE_NUMBER, eval=TRUE, echo=FALSE, results = "asis"} 
knitr::asis_output(paste0("Module name: ", names(cluster_set[MODULE_NUMBER, ]@gene_clusters)))
``` 

```{r, level_1_MOD_MODULE_NUMBER, eval=TRUE, echo=FALSE, results = "asis"} 
knitr::asis_output("## Expression {.panelset}")
``` 

```{r, heatmap_2_MOD_MODULE_NUMBER, eval="module_heatmap" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("This section provides various information about gene module MODULE_NUMBER.")
``` 

```{r, spatial_0_MOD_MODULE_NUMBER, eval="module_spatial" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### Spatial expression ")
``` 

```{r, spatial_1_MOD_MODULE_NUMBER, eval="module_spatial" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("In this section we compute the average expression levels of the gene expression program (i.e cluster) using Seurat::AddModuleScore(). This metagene is mapped onto spatial coordinates.")
``` 

```{r, spatial_2_MOD_MODULE_NUMBER, eval="module_spatial" %in% section, echo=FALSE, results = "asis"} 
SpatialFeaturePlot_used_params <- as.list(formals(Seurat::SpatialFeaturePlot))
for(i in names(SpatialFeaturePlot_params)){SpatialFeaturePlot_used_params[[i]] <- SpatialFeaturePlot_params[[i]]}

SpatialFeaturePlot_used_params$object <- seurat_object
SpatialFeaturePlot_used_params$features <- "MOD_MODULE_NUMBER"

base::do.call(Seurat::SpatialFeaturePlot, SpatialFeaturePlot_used_params)
``` 

```{r, heatmap_3_MOD_MODULE_NUMBER, eval="module_heatmap" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### Static heatmaps ")
``` 

```{r, heatmap_4_MOD_MODULE_NUMBER, eval="module_heatmap" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output("A heatmap with cells/spots as columns and genes as rows.") 
``` 

```{r heatmap_5_MOD_MODULE_NUMBER, echo=FALSE, eval="module_heatmap" %in% section }
plot_ggheatmap_used_params <- as.list(formals(scigenex::plot_ggheatmap))
for(i in names(plot_ggheatmap_params)){plot_ggheatmap_used_params[[i]] <- plot_ggheatmap_params[[i]]}

plot_ggheatmap_used_params$object <- cluster_set[MODULE_NUMBER,]
plot_ggheatmap_used_params$ident <- Idents(seurat_object)
base::do.call(scigenex::plot_ggheatmap, plot_ggheatmap_used_params)

```
<br>


```{r, iheatmap_2_MOD_MODULE_NUMBER, eval="module_iheatmap" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### Interactive heatmaps ")
``` 

```{r, iheatmap_3_MOD_MODULE_NUMBER, eval="module_iheatmap" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output(paste0("An interactive heatmap with cells/spots as columns and genes as rows. To limit the number of cells/spots displayed (which may  saturate the browser), the `subsample_by_ident()` function is used to randomly select a subset of cells/spots from each cluster. Here the maximum number of cells per population was : ", subsample_by_ident_params$nbcell, ".")) 
``` 

```{r iheatmap_4_MOD_MODULE_NUMBER, echo=FALSE, eval="module_iheatmap" %in% section }
subsample_by_ident_used_params <- as.list(formals(scigenex::subsample_by_ident))
for(i in names(subsample_by_ident_params)){subsample_by_ident_used_params[[i]] <- subsample_by_ident_params[[i]]}
subsample_by_ident_used_params$object <- cluster_set[MODULE_NUMBER,]
subsample_by_ident_used_params$ident <- Seurat::Idents(seurat_object)

plot_heatmap_used_params <- as.list(formals(scigenex::plot_heatmap))
for(i in names(plot_heatmap_params)){plot_heatmap_used_params[[i]] <- plot_heatmap_params[[i]]}

plot_heatmap_used_params$object <- base::do.call(scigenex::subsample_by_ident, subsample_by_ident_used_params)
plot_heatmap_used_params$cell_clusters <- Idents(seurat_object)
base::do.call(scigenex::plot_heatmap, plot_heatmap_used_params)

```

```{r, iumap_2_MOD_MODULE_NUMBER, eval="module_umap" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### UMAP Mapping ")
``` 

```{r, iumap_3_MOD_MODULE_NUMBER, eval="module_umap" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output("A UMAP in which mean gene module intensity is mapped to each cell/spot.") 
``` 

```{r iumap_4_MOD_MODULE_NUMBER, echo=FALSE, eval="module_umap" %in% section }

FeaturePlot_used_params <- as.list(formals(Seurat::FeaturePlot))
FeaturePlot_used_params <- FeaturePlot_used_params[!names(FeaturePlot_used_params) %in%  "sort.cell"] # deprecated

for(i in names(FeaturePlot_params)){FeaturePlot_used_params[[i]] <- FeaturePlot_params[[i]]}

FeaturePlot_used_params$object <- seurat_object
FeaturePlot_used_params$features <- "MOD_MODULE_NUMBER"

base::do.call(Seurat::FeaturePlot, FeaturePlot_used_params)

```

```{r, violin_2_MOD_MODULE_NUMBER, eval="module_violin" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### Violon plot ")
``` 

```{r, violin_3_MOD_MODULE_NUMBER, eval="module_violin" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output("Display the distribution of gene module mean expression levels across cell/spot populations.") 
``` 

```{r violin_4_MOD_MODULE_NUMBER, echo=FALSE, eval="module_violin" %in% section }
Seurat::VlnPlot(seurat_object, feature="MOD_MODULE_NUMBER")
```


```{r, gene_info_1_MOD_MODULE_NUMBER, eval="module_genes" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### Gene list ")
``` 

```{r, gene_info_2_MOD_MODULE_NUMBER, eval="module_genes" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output("The genes related to that module.") 
``` 

```{r gene_info_3_MOD_MODULE_NUMBER, echo=FALSE, eval="module_genes" %in% section}
df <- data.frame(Module=setNames(scigenex::gene_cluster(cluster_set[MODULE_NUMBER,]), NULL),
         gene=names(scigenex::gene_cluster(cluster_set[MODULE_NUMBER,])))
```

```{r gene_info_4_MOD_MODULE_NUMBER, echo=FALSE, eval="module_genes" %in% section}
DT::datatable(
    df,
    extensions = 'Buttons', 
    options = list(
        dom = 'Bfrtip',
        buttons =
            list(list(
                extend = 'csv',
                title = "gene_list_MOD_MODULE_NUMBER"))))
```

```{r, level_2_MOD_MODULE_NUMBER, eval=TRUE, echo=FALSE, results = "asis"} 
knitr::asis_output("## Cell Annotation {.panelset}")
``` 

```{r, cell_annot_IA_1_MOD_MODULE_NUMBER, eval="module_cell_annot_IA" %in% section, echo=FALSE, results = "asis"} 
knitr::asis_output("### IA-based annotation")
``` 

```{r, cell_annot_IA_2_MOD_MODULE_NUMBER, eval="module_cell_annot_IA" %in% section, echo=FALSE, results = "asis"}
knitr::asis_output("What Gemini tells us about the likely cell identity or underlying function of this gene module:") 
``` 

```{r, cell_annot_IA_3_MOD_MODULE_NUMBER, echo=FALSE, eval="module_cell_annot_IA" %in% section }
prompt <- paste0("\n\nGiven that you are a biologist working with ", smp_region, " cell compartment from ", smp_organ, " at ",  smp_stage , " stage. ",
                  "This biological sample was analyzed using transcriptomic analysis.",
                  " You found a list of co-regulated genes that contains :",
                  paste0(unlist(cluster_set[MODULE_NUMBER,]@gene_clusters), collapse=", "),
                  ". What could be the cells, functions or pathways most likely related to theses genes ? The answers (n=4 in total) should be ordered by likelyhood and formatted as a list with the following item : 'infered cell type/function/pathway' (#1, #2, #3, #4) and 'rational'\n\n.", collapse=" ")

knitr::asis_output("#### Prompt")

knitr::asis_output(prompt)

knitr::asis_output("#### IA Answer")

suppressMessages(gemini.R::setAPI("APIKEY"))
answer <- NULL
try_err <- try(suppressMessages(gemini.R::gemini_chat(prompt)))

if(inherits(try_err, "try-error")){
   cat("\n\nSeems we have an issue with Gemini...\n\n") 
}else{
  if(!is.null(try_err) )
    knitr::asis_output(try_err$outputs )
}

cat("\n\n")
```


```{r, func_anno_MOD_MODULE_NUMBER, eval=TRUE, echo=FALSE, results = "asis"}

for(ann in annotation_src){
 
  cat(paste0("\n\n## Functional Enrichment (", ann ,") {.panelset}\n\n"))
  
  if("edo" %in% ls()){rm(edo)}
  
  gene_set <- setNames(unlist(cluster_set[MODULE_NUMBER,]@gene_clusters), NULL)
  
  # Some gene may have a version number at the end, e.g. "gene1~2" (e.g. in the case
  # of a clusterSet created from Seurat::FindAllMarkers() output).
  gene_set <- gsub(gene_set, pattern = "~[0-9]+$", replacement = "")
  try(suppressMessages(query_entrezid <- AnnotationDbi::select(get(bioc_org_db), 
                                                               keys = gene_set,
                                                               columns = c("ENTREZID", "SYMBOL"),
                                                               keytype = "SYMBOL")))
  
  try(enrich_res <- clusterProfiler::enrichGO(query_entrezid[,"ENTREZID"],
                                             OrgDb = bioc_org_db,
                                             ont = ann,
                                             readable = TRUE), silent = TRUE)
  
  if(!"enrich_res" %in% ls()){enrich_res <- data.frame()}
  
  try(edo <- enrichplot::pairwise_termsim(enrich_res, 
                                          method = "Resnik", 
                                          showCategory = 20, 
                                          semData = sem_sim[[ann]]), silent = TRUE)
  
  if(!"edo" %in% ls()){edo <- data.frame()}

  
  if("module_term_network" %in% section){
    cat("\n\n### Cnetplot (fdp)\n\n")
    cat("\n\nThe cnetplot() function from the clusterProfiler R package is used to visualize enriched functional categories (e.g., here GO BP) as a concept-gene network. It helps interpret the relationships between genes and the biological functions or pathways they are associated with. The cnetplot() function generates a bipartite network plot where nodes represent genes and terms (e.g., GO terms, pathways). Edges connect genes to the terms they are annotated with. The network helps illustrate how multiple terms share common genes, and how individual genes may be involved in multiple functions.\n\n") 
    
    cnetplot_used_params <- as.list(formals(enrichplot:::cnetplot.enrichResult))
  for(i in names(cnetplot_params)){cnetplot_used_params[[i]] <- cnetplot_params[[i]]}
  # These guys are deprecated...
  cnetplot_used_params <- cnetplot_used_params[!names(cnetplot_used_params) %in% c("foldChange", 
                                                                                   "colorEdge",
                                                                                   "color_category",
                                                                                   "color_gene",
                                                                                   "cex_category",
                                                                                   "cex_gene",
                                                                                   "cex_label_category",
                                                                                   "cex_label_gene")]
  cnetplot_used_params$x <- edo
  cnetplot_used_params$circular <- FALSE
  
  if(nrow(edo) > 0){
    
    try_err <- try(base::do.call(enrichplot:::cnetplot.enrichResult, cnetplot_used_params) + 
                     Seurat::NoLegend(), silent = TRUE)
    
    if(inherits(try_err, "try-error")){
       cat("\n\nError when creating diagram.\n\n") 
    }else{
      print(try_err)
    }
  }else{
    cat("\n\nNo diagram available.\n\n") 
  }
  
  }
  
    if("module_term_network_circ" %in% section){
    cat("\n\n### Cnetplot (circular)\n\n")
    cat("\n\nThe cnetplot() function from the clusterProfiler R package is used to visualize enriched functional categories (e.g., here GO BP) as a concept-gene network. It helps interpret the relationships between genes and the biological functions or pathways they are associated with. The cnetplot() function generates a bipartite network plot where nodes represent genes and terms (e.g., GO terms, pathways). Edges connect genes to the terms they are annotated with. The network helps illustrate how multiple terms share common genes, and how individual genes may be involved in multiple functions.\n\n") 
  
    cnetplot_used_params <- as.list(formals(enrichplot:::cnetplot.enrichResult))
    for(i in names(cnetplot_params)){cnetplot_used_params[[i]] <- cnetplot_params[[i]]}
    # These guys are deprecated...
    cnetplot_used_params <- cnetplot_used_params[!names(cnetplot_used_params) %in% c("foldChange", 
                                                                                     "colorEdge",
                                                                                     "color_category",
                                                                                     "color_gene",
                                                                                     "cex_category",
                                                                                     "cex_gene",
                                                                                     "cex_label_category",
                                                                                     "cex_label_gene")]
    cnetplot_used_params$x <- edo
    cnetplot_used_params$circular <- TRUE
    
    if(nrow(edo) > 0){
      
      try_err <- try(base::do.call(enrichplot:::cnetplot.enrichResult, cnetplot_used_params) + 
                       Seurat::NoLegend(), silent = TRUE)
      
      if(inherits(try_err, "try-error")){
         cat("\n\nError when creating diagram.\n\n") 
      }else{
        print(try_err)
      }
    }else{
      cat("\n\nNo diagram available.\n\n") 
    }
    
    }
  
  
    if("module_term_barplot_1" %in% section){
      
      cat("\n\n### Func. enr. barplot \n\n")
      cat("\n\nDotplot of enriched terms.\n\n") 
      
    if(!is.null(enrich_res)){
      if(nrow(enrich_res) > 0){
        try_err <- try(dplyr::mutate(enrich_res, 
                                   qscore = -log(p.adjust, base=10)) %>% 
                                   barplot(x="qscore", showCategory=15) + 
                        ggplot2::theme(axis.text.y = ggplot2::element_text(size=7)), silent = TRUE)
  
        
        if(inherits(try_err, "try-error")){
           cat("\n\nError when creating diagram.\n\n") 
        }else{
          print(try_err)
        }
      }else{
      cat("\n\nNo diagram available.\n\n") 
     }
    }else{
      cat("\n\nNo diagram available.\n\n") 
    }
  }

  if("module_term_barplot_2" %in% section){
    
    cat("\n\n### Func. enr. dotplot \n\n")
    cat("\n\nDotplot of enriched terms.\n\n") 
    
    if(!is.null(enrich_res)){
      if(nrow(enrich_res) > 0){
                try_err <- try(enrichplot::dotplot(enrich_res, showCategory=15) + 
                     ggplot2::theme(axis.text.y = ggplot2::element_text(size=7)), silent = TRUE)
        
        if(inherits(try_err, "try-error")){
           cat("\n\nError when creating diagram.\n\n") 
        }else{
          print(try_err)
        }
      }else{
      cat("\n\nNo diagram available.\n\n") 
     }
    }else{
      cat("\n\nNo diagram available.\n\n") 
    }
  }
  
  
  if("term_table" %in% section){
    cat("\n\n### Func. enr. table \n\n")
    cat("\n\nA table with various information related to enrichment.\n\n") 
   
    font.size <- "50%"
    if(nrow(edo) > 0){
      
      df_edo <- edo@result
      if(all(is.numeric(df_edo$RichFactor)))
        df_edo$RichFactor <- round(df_edo$RichFactor, 4)
      if(all(is.numeric(df_edo$FoldEnrichment)))
        df_edo$FoldEnrichment <- round(df_edo$FoldEnrichment, 4)
      if(all(is.numeric(df_edo$zScore)))
        df_edo$zScore <- round(df_edo$zScore, 4)
      if(all(is.numeric(df_edo$pvalue)))  
        df_edo$pvalue <- formatC(df_edo$pvalue, format = "e", digits = 2)
      if(all(is.numeric(df_edo$p.adjust)))  
        df_edo$p.adjust <- formatC(df_edo$p.adjust, format = "e", digits = 2)
      if(all(is.numeric(df_edo$qvalue)))   
      df_edo$qvalue <- formatC(df_edo$qvalue, format = "e", digits = 2)
      rownames(edo@result) <- NULL
    

      # from https://tinyurl.com/mwxj7544
      print(htmltools::tagList(DT::datatable(df_edo,
                                             options=list(
                                             initComplete = htmlwidgets::JS(
                                                        "function(settings, json) {",
                                                        paste0("$(this.api().table().container()).css({'font-size': '", 
                                                               font.size, "'});"),
                                                        "}"),
                                             dom = 'Bfrtip',
                                             buttons = list(list(extend = 'csv',
                                                             title = "gene_list_MOD_MODULE_NUMBER"))
                       ))))
    
    }else{
      cat("\n\nNo table available.\n\n") 
    }
        
  }
}



``` 

