---
author: "Julie Bavais & Denis Puthier"
date: "`r Sys.Date()`"
title: "scRNA-seq guided tutorial"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{scRNA-seq guided tutorial}
  %\usepackage[UTF-8]{inputenc}
---

# Guided tutorial

The easiest way to use scigenex is to perform the following steps using the Seurat R package:

-   Load data into a Seurat object
-   Perform quality control
-   Perform normalization

The resulting object can be used as input to SciGeneX. You can also provide a normalized matrix as input.

## The dataset

For this tutorial, we'll be using a peripheral blood mononuclear cell (PBMC) scRNA-seq dataset available from the 10x Genomics website. This dataset contains 2700 individual cells sequenced on the Illumina NextSeq 500 and can be downloaded [**from 10X Genomics**](https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz) web site or via the `SeuratData` library.

### Preparing the pbmc3k dataset

In this step, we'll carry out the classic pre-processing steps of the tutorial. Please refer to [this tutorial](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html) for more information. If you have already pre-processed your data with Seurat, or if you have a normalized count matrix as input, you can skip this step.

```{r loading1, echo=FALSE, eval=TRUE, warning=FALSE, results='hide', message=FALSE}
suppressMessages(suppressWarnings(library(ggplot2, quietly = TRUE)))
suppressMessages(suppressWarnings(library(patchwork, quietly = TRUE)))
suppressMessages(suppressWarnings(library(Seurat, quietly = TRUE)))
suppressMessages(suppressWarnings(library(SeuratData, quietly = TRUE)))
suppressMessages(suppressWarnings(InstallData("pbmc3k", quiet=TRUE)))
suppressMessages(suppressWarnings(data("pbmc3k")))
pbmc3k <- UpdateSeuratObject(object = pbmc3k)

```

```{r loading2, echo=TRUE, eval=FALSE}
library(ggplot2)
library(patchwork)
library(Seurat, quietly = TRUE)
library(SeuratData)

InstallData("pbmc3k")
data("pbmc3k")
pbmc3k <- UpdateSeuratObject(object = pbmc3k)

```

We next run the classical steps of the seurat pipeline. For more information, you can check Seurat website [here](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html).

```{r classicalsteps, echo=TRUE, eval=TRUE, warning=FALSE}
# Quality control
pbmc3k[["percent.mt"]] <- PercentageFeatureSet(pbmc3k, pattern = "^MT-")
pbmc3k <- subset(pbmc3k, subset = percent.mt < 5 & nFeature_RNA > 200)

# Normalizing
pbmc3k <- NormalizeData(pbmc3k)

# Identification of highly variable genes
pbmc3k <- FindVariableFeatures(pbmc3k, selection.method = "vst", nfeatures = 2000)

# Scaling data
pbmc3k <- ScaleData(pbmc3k, features = rownames(pbmc3k), verbose = FALSE)

# Perform principal component analysis
pbmc3k <- RunPCA(pbmc3k, features = VariableFeatures(object = pbmc3k), verbose = FALSE)

# Cell clustering
pbmc3k <- FindNeighbors(pbmc3k, dims = 1:10, verbose = FALSE)
pbmc3k <- FindClusters(pbmc3k, resolution = 0.5, verbose = FALSE)

# Dimension reduction
pbmc3k <-suppressWarnings(RunUMAP(pbmc3k, dims = 1:10, verbose = FALSE))
dim_plot_orig <- DimPlot(pbmc3k, reduction = "umap")
dim_plot_orig
```

## Extracting gene modules using SciGeneX

In this section, we use the previously generated Seurat object as input to run the main SciGeneX commands. This command executes the main algorithm which will:

-   identify and extract co-expressed genes
-   divide the selected genes into groups
-   store the result in a ClusterSet object.

First, we'll load the SciGeneX library. To limit the verbosity of the SciGeneX functions, we'll set the verbosity level to zero (which will disable information and debug messages).

```{r loadlib1, warning=FALSE, results='hide', echo = FALSE, eval=TRUE}
suppressMessages(suppressWarnings(library(scigenex)))
scigenex::set_verbosity(0)
```

```{r loadlib2, echo = TRUE, eval=FALSE, warning=FALSE, results='hide'}
library(scigenex)
scigenex::set_verbosity(0)
```

Then we call successively the `select_genes()` function which will select informative genes (*i.e* co-regulated), then the `gene_clustering()` function which will call MCL and partition the dataset into gene modules.

```{r scigenex, echo=TRUE, eval=TRUE, results='hide', warning=FALSE }

# Select informative genes
pbmc_scigenex <- select_genes(pbmc3k,
                              k = 50,
                              distance_method = "pearson",
                              layer = "data",
                              row_sum=5)

# Run MCL
pbmc_scigenex <- gene_clustering(pbmc_scigenex,
                                 s = 5,
                                 threads = 2,
                                 inflation = 2)
```

The object produced is a `ClusterSet` objet that is a S4 object that is intented to store gene modules.

```{r isS4}
isS4(pbmc_scigenex)
pbmc_scigenex
```

There are various methods associated with the `ClusterSet` objects.

```{r show_methods}
show_methods()
```

The current object contains `r nrow(pbmc_scigenex)` informative genes, `r ncol(pbmc_scigenex)` samples and `r nclust(pbmc_scigenex)` gene modules.

```{r nrow_ncol}
nrow(pbmc_scigenex)
ncol(pbmc_scigenex)
nclust(pbmc_scigenex)
```

At this stage, several modules need to be filtered, as many of them may be singletons. Interestingly, the `ClusterSet` class implements the indexing operator/function ("["). The first argument/dimension of the indexing function corresponds to the cluster stored in the object. The second dimension corresponds to the cell/spot names. As an example, we can simply store gene modules whose size (\*i.e. number of genes) is greater than 7 using the following code. The result is an object containing gene modules `r nclust(pbmc_scigenex[clust_size(pbmc_scigenex) > 7, ])`.

```{r clust_size}
pbmc_scigenex <- pbmc_scigenex[clust_size(pbmc_scigenex) > 7, ]
nclust(pbmc_scigenex)
```

It may also be important to filter out gene based on dispersion. Several parameters can be computed for each cluster using the `cluster_stats()`function.

```{r plot_cluster_stats_1}
plot_cluster_stats(cluster_stats(pbmc_scigenex)) + 
 ggplot2::theme(axis.text.y = ggplot2::element_blank(),
                axis.text.x = element_text(angle=45, vjust = 0.5),
                panel.grid = element_blank()) 

```

Here we will select gene modules based on standard deviation (\> 0.1) and rename the cluster (from 1 to the number of clusters):

```{r rename_clust}
pbmc_scigenex <- pbmc_scigenex[cluster_stats(pbmc_scigenex)$sd > 0.1, ] 
pbmc_scigenex <- rename_clust(pbmc_scigenex)
nclust(pbmc_scigenex)
```

Then we check the statistics again.

```{r plot_cluster_stats}
plot_cluster_stats(cluster_stats(pbmc_scigenex)) + 
 ggplot2::theme(axis.text.y = ggplot2::element_blank(),
                axis.text.x = element_text(angle=45, vjust = 0.5),
                panel.grid = element_blank()) 
```

Clusters of genes are stored in the `gene_clusters` slot. One can access the gene names from a cluster using the `get_genes()` command. By default, all genes are returned.

```{r genes_module_5}
# Extract gene names from the 5th gene cluster
genes_module_5 <- get_genes(pbmc_scigenex, cluster = 5)
head(genes_module_5)
```

One can also access the gene to cluster mapping using `get_genes()`.

```{r head}
head(gene_cluster(pbmc_scigenex))
tail(gene_cluster(pbmc_scigenex))
```

## Heatmap visualization

### Visualization of specific genetic modules

Visualizing a heatmap containing all cells and modules can be time-consuming and require significant memory resources. A first alternative is to examine gene modules individually or a subset of modules. Gene modules can be visualized using the `plot_heatmap()` the `plot_ggheatmap()` functions. The former is primarily intended to provide an interactive visualization based on the `iheatmapr` library, and allows easy browsing of results and zooming in on particular regions of the heatmap. This second solution leads to a diagram that is more easily customized, as it is based on the ggplot framework. Here, we use `plot_ggheatmap()` to view the first 4 clusters. Note that here, we choose to order the columns/cells on the results of `Seurat::FindClusters`.

```{r plot_ggheatmap_1}
plot_ggheatmap(pbmc_scigenex[1:4, ], 
               use_top_genes = FALSE,
               ident=Idents(pbmc3k)) + ggtitle("Cluster 1 to 4") 
```

### Heatmap of representative genes

However, an alternative is to extract the most representative genes from each group. This can be achieved using the `top_genes()` function. This function stores the identifiers of these representative genes in the `top_genes` slot of the `ClusterSet` object. The `get_genes()` function is used to access the `top_genes` slot.

```{r top_genes}
pbmc_scigenex <- top_genes(pbmc_scigenex)
genes_cl5_top <- get_genes(pbmc_scigenex, cluster = 5, top = TRUE)
genes_cl5_top
```

Both `plot_heatmap()` and `plot_ggheatmap()` support the `use_top_genes`argument:

```{r plot_ggheatmap, warning=FALSE }

plot_ggheatmap(pbmc_scigenex, 
               use_top_genes = TRUE,
               ident=Seurat::Idents(pbmc3k)) + ggtitle("All clusters (top genes)") +
               theme(strip.text.y = element_text(size=4))
```

### Interactive heatmap

A very interesting feature of SciGeneX is its ability to display gene expression levels in cells/spots using interactive heatmaps. With this function, the user can interactively evaluate expression levels in selected cells or groups, or over the whole dataset. However, it is generally advisable, when using all clusters, to restrict the analysis using `top_genes=TRUE`. Here we will also select a subset of cells from each cluster.

```{r plot_heatmap, echo=TRUE, eval=TRUE}
sub_clust <- subsample_by_ident(pbmc_scigenex, 
                                        nbcell=15,
                                        ident=Seurat::Idents(pbmc3k)) 
                                        
plot_heatmap(sub_clust,
            use_top_genes = TRUE,
            cell_clusters =Seurat::Idents(pbmc3k))
```

## Improving cell resolution on UMAP

On can use scigenex selected genes as input for PCA. One can expect some improvement regarding cell population resolution compare to original Seurat pipeline :

```{r}
# Scaling data using genes from scigenex
pbmc3k <- ScaleData(pbmc3k, features = row_names(pbmc_scigenex), verbose = FALSE)

# Perform principal component analysis
pbmc3k <- RunPCA(pbmc3k, features = VariableFeatures(object = pbmc3k), verbose = FALSE)

# Cell clustering
pbmc3k <- FindNeighbors(pbmc3k, dims = 1:10, verbose = FALSE)
pbmc3k <- FindClusters(pbmc3k, resolution = 0.5, verbose = FALSE)

# Dimension reduction
pbmc3k <-suppressWarnings(RunUMAP(pbmc3k, dims = 1:10, verbose = FALSE))
dim_plot_sci <- DimPlot(pbmc3k, reduction = "umap")

dim_plot_orig + dim_plot_sci
```

## Exporting modules

Gene modules can be exported using the `cluster_set_to_xls()`. This function will create a Excel workbook that will contain the mapping from genes to modules. You may also use `write_clust()`.

```{r exporting_1, echo=FALSE}
tmp_file <- file.path(tempdir(), "pbmc_scigenex_out.xls")
out <- suppressWarnings(file.remove(tmp_file))
dir.create(dirname(tmp_file), recursive = TRUE, showWarnings = FALSE)
```

```{r exporting_2}
tmp_file <- file.path(tempdir(), "pbmc_scigenex_out.xls")
cluster_set_to_xls(object=pbmc_scigenex, file_path=tmp_file)

```

## Functional enrichment analysis

Functional enrichment analysis can be performed for each gene module using the `enrich_go()` function. Enrichments can be displayed using the `plot_clust_enrichments()` function.

```{r func_enrich, eval=TRUE, echo=TRUE}
# Functional enrichment analysis
pbmc_scigenex <- enrich_go(pbmc_scigenex, specie = "Hsapiens", ontology = "BP")
plot_clust_enrichments(pbmc_scigenex, gradient_palette=colors_for_gradient("Je1"), 
                       floor=50,
                       nb_go_term = 2) + 
  theme(axis.text.y = element_text(size=4))
```

## Mapping cell populations markers onto the gene modules

Given a set of markers, the `plot_markers_to_clusters()` function can be used to map cell type markers to gene modules. This function will display jaccard and hypergeometric statistics. Here we will use the markers from the [sctype.app](https://sctype.app/) database.

```{r markers, eval=FALSE, echo=TRUE}
library(dplyr)
library(tidyr)
```

```{r suppressMessages, eval=TRUE, echo=FALSE}
suppressMessages(suppressWarnings(library(dplyr)))
suppressMessages(suppressWarnings(library(tidyr)))
```

```{r plot_markers_to_clusters}
sctype <- "https://zenodo.org/record/8269433/files/sctype.app.tsv"
marker_table <- read.table(sctype, head=TRUE, sep="\t")
marker_table <- marker_table %>% 
                filter(Tissue == "Immune system") %>% 
                filter(!grepl('Pro-|Pre-|HSC|precursor|Progenitor', Cell.type)) %>%
                separate_rows(Marker_genes, convert = TRUE)

markers <- split(marker_table$Marker_genes, 
                 marker_table$Cell.type)
plot_markers_to_clusters(pbmc_scigenex, 
                         markers=markers, background = rownames(pbmc3k))
```

## Using scigenex selected genes to improve UMAP

One can restrict PCA computation to the subset of genes selected by Scigenex. One can expected a significant improvement of segregation of cell populations. To do so one needs to provide the list of genes selected by scigenex to the `ScaleData()` function of Seurat. A indicated in the RunPCA() help function : "PCA will be run using the variable features for the Assay. Note that the features must be present in the scaled data. Any requested features that are not scaled or have 0 variance will be dropped, and the PCA will be run using the remaining features."

```{r improve_umap}
# Scaling data
feature_2_scale <- row_names(pbmc_scigenex)
pbmc3k_alt <- ScaleData(pbmc3k, features = feature_2_scale, verbose = FALSE)

# Perform principal component analysis
pbmc3k_alt <- RunPCA(pbmc3k_alt, features = VariableFeatures(object = pbmc3k_alt), verbose = FALSE)

# Cell clustering
pbmc3k_alt <- FindNeighbors(pbmc3k_alt, dims = 1:10, verbose = FALSE)
pbmc3k_alt <- FindClusters(pbmc3k_alt, resolution = 0.5, verbose = FALSE)

# Dimension reduction
pbmc3k_alt <-suppressWarnings(RunUMAP(pbmc3k_alt, dims = 1:10, verbose = FALSE))

# Compare alternative projection (pbmc3k_alt) to original (pbmc3k)
DimPlot(pbmc3k, reduction = "umap")  + DimPlot(pbmc3k_alt, reduction = "umap") 
```

## Creating a report

A report can be created using the `cluster_set_report()` function. The arguments to this function are the processed clusterSet and the corresponding processed Seurat object. Ideally, the clusterSet object should contain functional annotations (see `enrich_go()`). Currently, the process of creating a report can be quite time consuming It can also produce heavy html files which may take some time to load in the web browser.

```{r report, echo=TRUE, eval=FALSE, warning=FALSE, results='hide'}
# Uncomment to prepare the report.
# gcss_brain <- enrich_go(gcss_brain, species = "Mmusculus", ontology = "BP")
# cluster_set_report(clusterset_object = gcss_brain, seurat_object = brain1)
```

## Session info

```{r}
sessionInfo()
```
