<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css"><script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet"><script src="pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-title-body">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Changelog</h1>
    </div>

<div id="changelog" class="section level1">

<div class="section level2">
<h2 id="v152">v1.5.2<a class="anchor" aria-label="anchor" href="#v152"></a></h2>
<ul><li>Fix #147 and #145.</li>
<li>Added cmp_to_a_list() function.</li>
<li>Added coord_flip argument to plot_ggheatmap()</li>
<li>Added top_by_grep()</li>
<li>Added as_list argument to grep_clust().</li>
<li>Added single_file to write_clust().</li>
<li>Added to_lin to plot_profile.</li>
<li>Deleted dot_prod from cluster_stats() and added CV.</li>
<li>The row_sum is now normalized by cluster size in cluster_stats().</li>
<li>Added top_by_intersect().</li>
</ul></div>
<div class="section level2">
<h2 id="v151">v1.5.1<a class="anchor" aria-label="anchor" href="#v151"></a></h2>
<ul><li>Fix gene_clustering() and mcl_install(). Added ‘mcl_dir’ argument to <code><a href="reference/gene_clustering.html">gene_clustering()</a></code> (in case automated installation in difficult, e.g., when running in a proprietary environnement).</li>
<li>Fix <code><a href="reference/top_by_go.html">top_by_go()</a></code>. Added argument ‘as_list’.</li>
<li>Documentation has been splitted in two part (scRNA-seq and ST).</li>
</ul></div>
<div class="section level2">
<h2 id="v1413">v1.4.13<a class="anchor" aria-label="anchor" href="#v1413"></a></h2>
<ul><li>Added support for Louvain and walktrap: call_walktrap_clusterset(), call_louvain_clusterset()</li>
<li>construct_new_graph() was renamed do_closest_neighbor_graph()</li>
<li>keep_dbf_graph() was renamed do_reciprocal_neighbor_graph()</li>
<li>code refactoring (mainly comments).</li>
<li>Deleted code relics (ident argument as list) from plot_ggheatmap().</li>
<li>Added the subsample_by_ident() method for a ClusterSet object. handy to be used with plot_ggheatmap or plot_heatmap (to put emphasis on small populations).</li>
<li>Added grep_clust() to search for genes in a ClusterSet object using a REGEXP.</li>
<li>Added reload_pac() (for development).</li>
</ul></div>
<div class="section level2">
<h2 id="v1410">v1.4.10<a class="anchor" aria-label="anchor" href="#v1410"></a></h2>
<ul><li>Fix NAMESPACE (xlxs dependency).</li>
<li>Rebuilt doc in html format.</li>
</ul></div>
<div class="section level2">
<h2 id="v149">v1.4.9<a class="anchor" aria-label="anchor" href="#v149"></a></h2>
<ul><li>Remove xlxs dependency in favor with WriteXLS dependency (to avoid rJava dependency).</li>
</ul></div>
<div class="section level2">
<h2 id="v148">v1.4.8<a class="anchor" aria-label="anchor" href="#v148"></a></h2>
<ul><li>Fix an issue with temp dir.</li>
</ul></div>
<div class="section level2">
<h2 id="v147">v1.4.7<a class="anchor" aria-label="anchor" href="#v147"></a></h2>
<ul><li><p>Change k_g argument to ‘s’ in <code><a href="reference/gene_clustering.html">gene_clustering()</a></code> function for clarity purpose.</p></li>
<li><p>Added dependency to DT library (required by <code><a href="reference/cluster_set_report.html">cluster_set_report()</a></code>).</p></li>
<li><p>Updated html doc.</p></li>
<li><p>Fix an issue with pandoc in <code>clusterset_report()</code>.</p></li>
<li><p>Improved importfrom in <code>clusterset_report()</code>.</p></li>
</ul></div>
<div class="section level2">
<h2 id="v146">v1.4.6<a class="anchor" aria-label="anchor" href="#v146"></a></h2>
<ul><li>Prefix version with a v…</li>
</ul></div>
<div class="section level2">
<h2 id="v145">v1.4.5<a class="anchor" aria-label="anchor" href="#v145"></a></h2>
<ul><li>keep_nn argument is deprecated (<code><a href="reference/gene_clustering.html">gene_clustering()</a></code>).</li>
</ul></div>
<div class="section level2">
<h2 id="v144">v1.4.4<a class="anchor" aria-label="anchor" href="#v144"></a></h2>
<ul><li>Fix #134</li>
</ul></div>
<div class="section level2">
<h2 id="v143">v1.4.3<a class="anchor" aria-label="anchor" href="#v143"></a></h2>
<ul><li><p>Remove DynmicTreeCut dependency.</p></li>
<li><p>Fix bug #137.</p></li>
</ul></div>
<div class="section level2">
<h2 id="v142">v1.4.2<a class="anchor" aria-label="anchor" href="#v142"></a></h2>
<ul><li><p>Added color for cell cluster in <code><a href="reference/plot_ggheatmap.html">plot_ggheatmap()</a></code>.</p></li>
<li><p>Updated doc.</p></li>
</ul></div>
<div class="section level2">
<h2 id="v141">v1.4.1<a class="anchor" aria-label="anchor" href="#v141"></a></h2>
<ul><li><p>Minor bug fix.</p></li>
<li><p>Updated doc</p></li>
</ul></div>
<div class="section level2">
<h2 id="v140">v1.4.0<a class="anchor" aria-label="anchor" href="#v140"></a></h2>
<ul><li>Update of the documentation</li>
<li>Added support for spatial transcriptomics: <code><a href="reference/getFlippedTissueCoordinates.html">getFlippedTissueCoordinates()</a></code>, <code><a href="reference/plot_spatial.html">plot_spatial()</a></code>, <code><a href="reference/plot_spatial_panel.html">plot_spatial_panel()</a></code>.</li>
<li>Added import from matrix and Seurat object: <code><a href="reference/cluster_set_from_matrix.html">cluster_set_from_matrix()</a></code>, <code><a href="reference/cluster_set_from_seurat.html">cluster_set_from_seurat()</a></code>.</li>
<li>Added <code><a href="reference/cluster_stats.html">cluster_stats()</a></code> and <code><a href="reference/plot_cluster_stats.html">plot_cluster_stats()</a></code> to perform statistics on clusters.</li>
<li>Added export: <code><a href="reference/cluster_set_to_xls.html">cluster_set_to_xls()</a></code>.</li>
<li>Added <code><a href="reference/compare_genesets.html">compare_genesets()</a></code> and <code><a href="reference/plot_cmp_genesets.html">plot_cmp_genesets()</a></code> to compare gene lists with various statistics.</li>
<li>Added <code><a href="reference/plot_clust_enrichments.html">plot_clust_enrichments()</a></code> to compare functional enrichment over clusters.</li>
<li>Added <code><a href="reference/plot_ggheatmap.html">plot_ggheatmap()</a></code> for statically display heatmaps.</li>
<li>Added <code><a href="reference/plot_markers_to_clusters.html">plot_markers_to_clusters()</a></code> to map markers onto clusters.</li>
<li>Added <code><a href="reference/top_by_go.html">top_by_go()</a></code> to select top genes by GO terms.</li>
</ul></div>
<div class="section level2">
<h2 id="v131">v1.3.1<a class="anchor" aria-label="anchor" href="#v131"></a></h2>
<ul><li>Update of the documentation</li>
<li>Improve get_genes function : gene names are ordered by gene clusters</li>
<li>Fix viz_enrich function : viz_enrich function is now working when using a specific ontology db as ontology parameter in enrich_go function</li>
</ul></div>
<div class="section level2">
<h2 id="v130">v1.3.0<a class="anchor" aria-label="anchor" href="#v130"></a></h2>
<ul><li>Add a clustering method using hierchical clustering and DynamicTreeCut R package</li>
<li>Add a cell_clusters slot in the ClusterSet object</li>
<li>Create a new documentation</li>
<li>Modification of the av_dot_prod_min filter : use the max of the median values of the dot product</li>
<li>Add get_genes and get_cells functions</li>
<li>Improve the computational speed of DBFMCL function</li>
<li>Replace the name of the slot “cluster” by “gene_patterns”</li>
<li>Implementation of new test (testthat)</li>
</ul></div>
<div class="section level2">
<h2 id="v1221">v1.2.21<a class="anchor" aria-label="anchor" href="#v1221"></a></h2>
<ul><li>Update the package used for enrichment analysis to use ClusterProfiler</li>
<li>Update visualization of enrichment analysis results : add dotplot and barplot provided by ClusterProfiler</li>
<li>Add more parameters to control the density plot of distances</li>
<li>Fix issues from Github</li>
</ul></div>
<div class="section level2">
<h2 id="v122">v1.2.2<a class="anchor" aria-label="anchor" href="#v122"></a></h2>
<ul><li>Replace the C code for DBF function by a R implementation to reduce the computational time.</li>
<li>Split the main R script into sub-script related to each function of scigenex package.</li>
</ul></div>
<div class="section level2">
<h2 id="v112">v1.1.2<a class="anchor" aria-label="anchor" href="#v112"></a></h2>
<ul><li>Create plot_heatmap function to generate an interactive heatmap of the gene clusters</li>
<li>Modify NAMESPACE and DESCRIPTION file and add import_package.R files to automatically update dependencies using roxygen2::roxygenise()</li>
<li>Create top_genes function to find the top n genes from each signature. This function is also usefull for the plot_heatmap function.</li>
<li>Add a slot in the ClusterSet object containing the filename of the output file including simulated distances and cutting threshold.</li>
<li>Modify the input of plot_dist function to a ClusterSet object.</li>
<li>Update enric_analysis function : run the functional enrichment analysis for each gene cluster.</li>
<li>Remove R implementation of MCL. The package used is no more maintained and the released version needs to be fixed.</li>
</ul></div>
<div class="section level2">
<h2 id="v102">v1.0.2<a class="anchor" aria-label="anchor" href="#v102"></a></h2>
<ul><li>Fix fprint_selected function in C code byt changing type of len varibale to size_t</li>
<li>Set default optional_output parameter of DBF and DBFMCL function to TRUE</li>
</ul></div>
<div class="section level2">
<h2 id="v101">v1.0.1<a class="anchor" aria-label="anchor" href="#v101"></a></h2>
<div class="section level3">
<h3 id="bug-fixes">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes"></a></h3>
</div>
<div class="section level3">
<h3 id="apicli-changes">API/CLI Changes<a class="anchor" aria-label="anchor" href="#apicli-changes"></a></h3>
<ul><li>Scigenex now depends on R 4.0.0</li>
<li>No more reference to Biobase in the code.</li>
</ul></div>
<div class="section level3">
<h3 id="code-changes">Code changes<a class="anchor" aria-label="anchor" href="#code-changes"></a></h3>
</div>
<div class="section level3">
<h3 id="new-features">New Features<a class="anchor" aria-label="anchor" href="#new-features"></a></h3>
</div>
</div>
<div class="section level2">
<h2 id="v017">v0.1.7<a class="anchor" aria-label="anchor" href="#v017"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1"></a></h3>
<ul><li>Fix call to MCL.</li>
</ul></div>
<div class="section level3">
<h3 id="apicli-changes-1">API/CLI Changes<a class="anchor" aria-label="anchor" href="#apicli-changes-1"></a></h3>
</div>
<div class="section level3">
<h3 id="code-changes-1">Code changes<a class="anchor" aria-label="anchor" href="#code-changes-1"></a></h3>
</div>
<div class="section level3">
<h3 id="new-features-1">New Features<a class="anchor" aria-label="anchor" href="#new-features-1"></a></h3>
</div>
</div>
<div class="section level2">
<h2 id="v016">v0.1.6<a class="anchor" aria-label="anchor" href="#v016"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-2">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-2"></a></h3>
</div>
<div class="section level3">
<h3 id="apicli-changes-2">API/CLI Changes<a class="anchor" aria-label="anchor" href="#apicli-changes-2"></a></h3>
<ul><li>Now provides a support for R MCL library (not recommended).</li>
</ul></div>
<div class="section level3">
<h3 id="code-changes-2">Code changes<a class="anchor" aria-label="anchor" href="#code-changes-2"></a></h3>
</div>
<div class="section level3">
<h3 id="new-features-2">New Features<a class="anchor" aria-label="anchor" href="#new-features-2"></a></h3>
</div>
</div>
<div class="section level2">
<h2 id="v014">v0.1.4<a class="anchor" aria-label="anchor" href="#v014"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-3">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-3"></a></h3>
</div>
<div class="section level3">
<h3 id="apicli-changes-3">API/CLI Changes<a class="anchor" aria-label="anchor" href="#apicli-changes-3"></a></h3>
<ul><li><p>The <em>DBFMCLresult</em> object has been replaced by a more versatile object: <em>ClusterSet</em>.</p></li>
<li><p>The <em>plot_dbf()</em> method has been replaced by <em>plot_clust()</em>.</p></li>
<li><p>The <em>write_dbf()</em> method has been replaced by <em>write_clust()</em>.</p></li>
</ul></div>
<div class="section level3">
<h3 id="code-changes-3">Code changes<a class="anchor" aria-label="anchor" href="#code-changes-3"></a></h3>
<ul><li>Several code changes which were not clearly traced at the moment… Sry.</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-3">New Features<a class="anchor" aria-label="anchor" href="#new-features-3"></a></h3>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

