<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Creating a report • scigenex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Creating a report">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
<li>
  <a href="../articles/report.html">3 - Creating a bookdown report</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Creating a report</h1>
                        <h4 data-toc-skip class="author">Denis
Puthier</h4>
            
            <h4 data-toc-skip class="date">2025-07-06</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/vignettes/report.Rmd"><code>vignettes/report.Rmd</code></a></small>
      <div class="hidden name"><code>report.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to generate a detailled gene module
reports using the <code><a href="../reference/gm_report.html">gm_report()</a></code> function from the
<strong>scigenex</strong> R package. The reports are created in
<strong>Bookdown</strong> format and provide an integrated view of gene
module analysis from any Seurat supported technology:</p>
<ul>
<li>Single-cell RNA-seq experiments</li>
<li>Spatial transcriptomics (ST) experiments (Visium, Xenium,
Merscope…)</li>
<li>Bulk RNA-seq experiments</li>
</ul>
<p>The <code><a href="../reference/gm_report.html">gm_report()</a></code> function combines cluster information,
functional annotation, and visualization into a shareable HTML report.
It also optionally leverages AI-based tools for cell-type
annotation.</p>
</div>
<div class="section level2">
<h2 id="what-you-need">What You Need<a class="anchor" aria-label="anchor" href="#what-you-need"></a>
</h2>
<p>Before running <code><a href="../reference/gm_report.html">gm_report()</a></code>, you must have:</p>
<ul>
<li>A <strong>ClusterSet</strong> object, representing gene
modules.</li>
<li>A corresponding <strong>Seurat</strong> object containing your
expression data, cell cluster annotation, spatial location…</li>
<li>Optional: An API key for Gemini AI to enhance cell-type
annotation.</li>
</ul>
</div>
<div class="section level2">
<h2 id="how-to-create-a-clusterset">How to Create a ClusterSet<a class="anchor" aria-label="anchor" href="#how-to-create-a-clusterset"></a>
</h2>
<p>A <code>ClusterSet</code> can be built in several ways:</p>
<ul>
<li>
<strong>From the SciGeneX pipeline</strong>
<ul>
<li>Run <code><a href="../reference/select_genes.html">select_genes()</a></code> and then
<code><a href="../reference/gene_clustering.html">gene_clustering()</a></code> on a Seurat object or matrix.</li>
<li>Use filtering functions (e.g. <code><a href="../reference/filter_cluster_size.html">filter_cluster_size()</a></code>,
<code><a href="../reference/filter_cluster_sd.html">filter_cluster_sd()</a></code>) to select modules based on gene
number, variance, or other criteria.</li>
<li>
<strong>Objective:</strong>
<ul>
<li>Discover <strong>new gene modules</strong> directly from
co-expression patterns in your data, without prior biological
assumptions.</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><br></p>
<ul>
<li>
<strong>From a Seurat object based on cell-specific marker
genes</strong>
<ul>
<li>Use <code><a href="../reference/cluster_set_from_seurat.html">cluster_set_from_seurat()</a></code> with a Seurat object and
the result from <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">Seurat::FindAllMarkers()</a></code>.</li>
<li>
<strong>Objective:</strong>
<ul>
<li>Construct gene modules based on <strong>known marker genes</strong>
identified in a typical Seurat pipelines <br>
</li>
</ul>
</li>
</ul>
</li>
<li>
<strong>From a Seurat object based on any markers of
interest</strong>
<ul>
<li>Use <code><a href="../reference/cluster_set_from_seurat.html">cluster_set_from_seurat()</a></code> with a Seurat object and
a named vector (clusters with gene_names as names).</li>
<li>
<strong>Objective:</strong>
<ul>
<li>Construct gene modules based on known or identified pathways.</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<p>To use the <code><a href="../reference/gm_report.html">gm_report()</a></code> function, you need to load the
<strong>scigenex</strong>, <strong>Seurat</strong>, and
<strong>org.Hs.eg.db</strong> packages. The following code snippet
demonstrates how to load these libraries.</p>
</div>
<div class="section level2">
<h2 id="some-examples">Some examples<a class="anchor" aria-label="anchor" href="#some-examples"></a>
</h2>
<div class="section level3">
<h3 id="running-gm_report-from-a-seurat-single-cell-and-corresponding-scigenex-modules">Running gm_report() from a Seurat single-cell and corresponding
Scigenex modules<a class="anchor" aria-label="anchor" href="#running-gm_report-from-a-seurat-single-cell-and-corresponding-scigenex-modules"></a>
</h3>
<p>We will demonstrate with built-in example datasets from the
<strong>scigenex</strong> package. - We will use the pbmc3k_medium
dataset, which is a subset of the pbmc3k dataset from SeuratData. - The
<code>pbmc3k_medium_clusters</code> ClusterSet stores the gene modules
and was created from the <code>pbmc3k_medium</code> object using the
SciGeneX pipeline.</p>
<p>The report will be saved in the folder specified by
<code>out_dir</code>. Open <code>index.html</code> in your browser to
view it.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Unset verbosity to avoid cluttering the output</span></span>
<span><span class="fu"><a href="../reference/set_verbosity.html">set_verbosity</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example gene module clusters and Seurat object</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium_clusters"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a temporary directory for the report output</span></span>
<span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run gm_report() to generate the report</span></span>
<span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">pbmc3k_medium_clusters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span>,</span>
<span>  smp_stage <span class="op">=</span> <span class="st">"adult"</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">pbmc3k_medium</span>,</span>
<span>  report_title <span class="op">=</span> <span class="st">"PBMC Gene Module Report"</span>,</span>
<span>  report_subtitle <span class="op">=</span> <span class="st">"PBMC 3k Medium Dataset"</span>,</span>
<span>  report_author <span class="op">=</span> <span class="st">"scigenex Vignette"</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"pbmc_report"</span><span class="op">)</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-seurat-markers">Running gm_report() from a Seurat spatial experiment and
corresponding Seurat markers<a class="anchor" aria-label="anchor" href="#running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-seurat-markers"></a>
</h3>
<p>As indicated previously, <code><a href="../reference/gm_report.html">gm_report()</a></code>can be run from a
Seurat object together with on cell-specific marker genes obtained from
<code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">Seurat::FindAllMarkers()</a></code>. This represent a convenient way
to create a Seurat analysis report. In this case we need first to run
the <code><a href="../reference/cluster_set_from_seurat.html">cluster_set_from_seurat()</a></code> function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Unset verbosity to avoid cluttering the output</span></span>
<span><span class="fu"><a href="../reference/set_verbosity.html">set_verbosity</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">pbmc3k_medium</span>, only.pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cluster_set_from_seurat.html">cluster_set_from_seurat</a></span><span class="op">(</span><span class="va">pbmc3k_medium</span>, <span class="va">markers</span>, p_val_adj <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a temporary directory for the report output</span></span>
<span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run gm_report() to generate the report</span></span>
<span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">cs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">pbmc3k_medium</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  api_key <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># Optional: Gemini API key for IA-based annotation</span></span>
<span>  report_title <span class="op">=</span> <span class="st">"PBMC Gene Module Report"</span>,</span>
<span>  report_subtitle <span class="op">=</span> <span class="st">"PBMC 3k Medium Dataset"</span>,</span>
<span>  report_author <span class="op">=</span> <span class="st">"scigenex Vignette"</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"pbmc_report"</span><span class="op">)</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-scigenex-modules">Running gm_report() from a Seurat spatial experiment and
corresponding Scigenex modules<a class="anchor" aria-label="anchor" href="#running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-scigenex-modules"></a>
</h3>
<p>When running the <code><a href="../reference/gm_report.html">gm_report()</a></code> function on spatial
transcriptomics data, you can visualize gene modules in the context of
tissue architecture. The function supports Seurat objects with spatial
data and can generate spatial plots. Here we will use a tiny subset of
the 10X genomics lymph_node VISIUM (V1) dataset.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a temporary directory for the report output</span></span>
<span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example gene module clusters and Seurat object for spatial data</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_clusters_2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run gm_report() for spatial transcriptomics data</span></span>
<span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">lymph_node_tiny_clusters_2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">lymph_node_tiny_2</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  is_spatial_exp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  SpatialFeaturePlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  SpatialDimPlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"spatial_report"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-seurat-markers-1">Running gm_report() from a Seurat spatial experiment and
corresponding Seurat markers<a class="anchor" aria-label="anchor" href="#running-gm_report-from-a-seurat-spatial-experiment-and-corresponding-seurat-markers-1"></a>
</h3>
<p>In the same way, we can apply <code><a href="../reference/gm_report.html">gm_report()</a></code>with spatial
transcriptomics data and markers identified using the
<code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">Seurat::FindAllMarkers()</a></code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a temporary directory for the report output</span></span>
<span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example gene module clusters and Seurat object for spatial data</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_clusters_2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run gm_report() for spatial transcriptomics data</span></span>
<span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">lymph_node_tiny_clusters_2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">lymph_node_tiny_2</span>,</span>
<span>  smp_species <span class="op">=</span> <span class="st">"Homo sapiens"</span>,</span>
<span>  smp_region <span class="op">=</span> <span class="st">"total"</span>,</span>
<span>  smp_organ <span class="op">=</span> <span class="st">"lymph node"</span>,</span>
<span>  smp_stage <span class="op">=</span> <span class="st">"adult"</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  is_spatial_exp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  SpatialFeaturePlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  SpatialDimPlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"spatial_report"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="customizing-your-report">Customizing Your Report<a class="anchor" aria-label="anchor" href="#customizing-your-report"></a>
</h2>
<p>The <code><a href="../reference/gm_report.html">gm_report()</a></code>function accepts many parameters to
tailor the analysis and appearance:</p>
<ul>
<li>Plot customization
<ul>
<li>plot_profiles_params</li>
<li>FeaturePlot_params</li>
<li>SpatialDimPlot_params</li>
<li>…</li>
</ul>
</li>
<li>Selective content
<ul>
<li>Enable or disable specific report sections with the section
argument.</li>
</ul>
</li>
<li>AI-based annotation
<ul>
<li>Provide an <code>api_key</code>for cell-type predictions via Gemini
AI.</li>
</ul>
</li>
<li>Spatial transcriptomics
<ul>
<li>Set <code>is_spatial_exp = TRUE</code> if working with ST data.</li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The <code><a href="../reference/gm_report.html">gm_report()</a></code> function in Scigenex offers a powerful
way to analyze and document gene modules using Seurat and clusterSet
objects. It supports spatial transcriptomics, functional annotation, and
AI-based interpretation.</p>
<p>For advanced use, explore:</p>
<ul>
<li>
<code><a href="../reference/gm_report.html">help(gm_report)</a></code> in R</li>
<li>Scigenex documentation</li>
<li>Customizing Bookdown templates via <code>rmd_dir</code>
argument.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
