<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Creating a report â€¢ scigenex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Creating a report">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
<li>
  <a href="../articles/report.html">3 - Creating a bookdown report</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Creating a report</h1>
                        <h4 data-toc-skip class="author">Denis
Puthier</h4>
            
            <h4 data-toc-skip class="date">2025-07-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/vignettes/report.Rmd"><code>vignettes/report.Rmd</code></a></small>
      <div class="hidden name"><code>report.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="writing-report-about-clusterset-objects">Writing report about ClusterSet Objects<a class="anchor" aria-label="anchor" href="#writing-report-about-clusterset-objects"></a>
</h2>
<p>This vignette illustrates how to use the <code><a href="../reference/gm_report.html">gm_report()</a></code>
function from the scigenex R package to create a comprehensive gene
module analysis report in Bookdown format. This function integrates
cluster information, functional annotation, and visualization,
optionally supporting spatial transcriptomics (ST) data.</p>
</div>
<div class="section level2">
<h2 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h2>
<p>We will use a tiny built-in examples from the Scigenex library.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dputhier/scigenex">scigenex</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'SeuratObject::JS' by 'htmlwidgets::JS' when</span></span>
<span><span class="co">## loading 'scigenex'</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: SeuratObject</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: sp</span></span></code></pre>
<pre><code><span><span class="co">## 'SeuratObject' was built under R 4.4.0 but the current version is</span></span>
<span><span class="co">## 4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI</span></span>
<span><span class="co">## for R may have changed</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'SeuratObject'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, t</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:SeuratObject':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'IRanges'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:sp':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     %over%</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set verbosity for debugging</span></span>
<span><span class="fu"><a href="../reference/set_verbosity.html">set_verbosity</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load example gene module clusters and Seurat object</span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium_clusters"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- INFO :  Dataset 7871581/files/pbmc3k_medium_clusters has been loaded.</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- INFO :  Dataset 7871581/files/pbmc3k_medium has been loaded.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="running-the-gm_report-function">Running the gm_report() function<a class="anchor" aria-label="anchor" href="#running-the-gm_report-function"></a>
</h2>
<p>The gm_report() function requires a ClusterSet and a Seurat object.
It will:</p>
<ul>
<li>Compute module scores (<em>i.e</em> Using Seurat
<code><a href="https://satijalab.org/seurat/reference/AddModuleScore.html" class="external-link">AddModuleScore()</a></code>).</li>
<li>Perform functional enrichment (GO).</li>
<li>Generate various diagnosis diagrams.</li>
<li>Optionally perform cell-type annotation via Gemini AI</li>
<li>Render the report using Bookdown.</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">pbmc3k_medium_clusters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">pbmc3k_medium</span>,</span>
<span>  smp_species <span class="op">=</span> <span class="st">"Homo sapiens"</span>,</span>
<span>  smp_region <span class="op">=</span> <span class="st">"total"</span>,</span>
<span>  smp_organ <span class="op">=</span> <span class="st">"blood"</span>,</span>
<span>  smp_stage <span class="op">=</span> <span class="st">"adult"</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  api_key <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># Optional: Gemini API key for IA-based annotation</span></span>
<span>  report_title <span class="op">=</span> <span class="st">"PBMC Gene Module Report"</span>,</span>
<span>  report_subtitle <span class="op">=</span> <span class="st">"PBMC 3k Medium Dataset"</span>,</span>
<span>  report_author <span class="op">=</span> <span class="st">"scigenex Vignette"</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"pbmc_report"</span><span class="op">)</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- DEBUG :  Checking seurat object. </span></span>
<span><span class="co">## |-- DEBUG :  Checking ClusterSet format. </span></span>
<span><span class="co">## |-- INFO :  This is not a ST experiment... </span></span>
<span><span class="co">## |-- INFO :  Canceling ST reporting module. </span></span>
<span><span class="co">## |-- INFO :  No Gemini key provided... </span></span>
<span><span class="co">## |-- INFO :  Canceling IA-based cell type annotation. </span></span>
<span><span class="co">## |-- INFO :  Computing module scores... </span></span>
<span><span class="co">## |-- DEBUG :  Created temporary directory: </span></span>
<span><span class="co">## |-- INFO :  /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-47-12-2025 </span></span>
<span><span class="co">## |-- DEBUG :  Copying file in temporary directory. </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: _bookdown.yml </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: footer.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: index.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: module.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: style.css </span></span>
<span><span class="co">## |-- DEBUG :  Computing top genes. </span></span>
<span><span class="co">## |-- DEBUG :  Checking ClusterSet format. </span></span>
<span><span class="co">## |-- DEBUG :  Using fast computation of pearson correlations. </span></span>
<span><span class="co">## |-- DEBUG :  Using fast computation of pearson correlations. </span></span>
<span><span class="co">## |-- DEBUG :  Using fast computation of pearson correlations. </span></span>
<span><span class="co">## |-- INFO :  Results are stored in 'top_genes' slot of the object. </span></span>
<span><span class="co">## |-- DEBUG :  Computing centers. </span></span>
<span><span class="co">## |-- INFO :  Computing centers. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing matrix. </span></span>
<span><span class="co">## |-- DEBUG :  Renaming columns / rows. </span></span>
<span><span class="co">## |-- DEBUG :  Looping over clusters... </span></span>
<span><span class="co">## |-- DEBUG :  Computing cluster 1 center. </span></span>
<span><span class="co">## |-- DEBUG :  Computing cluster 2 center. </span></span>
<span><span class="co">## |-- DEBUG :  Computing cluster 3 center. </span></span>
<span><span class="co">## |-- INFO :  Computing semantic similarity (CC) for cnet_plot. </span></span>
<span><span class="co">## |-- INFO :  Preparing parameters for the report. </span></span>
<span><span class="co">## |-- INFO :  Looping through parameterised Reports. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing Rmd files for modules 1 </span></span>
<span><span class="co">## |-- DEBUG :  file : /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-47-12-2025/rmarkdown/module_0001.Rmd </span></span>
<span><span class="co">## |-- DEBUG :  Preparation of Rmd files for objects 1 finished. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing Rmd files for modules 2 </span></span>
<span><span class="co">## |-- DEBUG :  file : /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-47-12-2025/rmarkdown/module_0002.Rmd </span></span>
<span><span class="co">## |-- DEBUG :  Preparation of Rmd files for objects 2 finished. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing Rmd files for modules 3 </span></span>
<span><span class="co">## |-- DEBUG :  file : /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-47-12-2025/rmarkdown/module_0003.Rmd </span></span>
<span><span class="co">## |-- DEBUG :  Preparation of Rmd files for objects 3 finished. </span></span>
<span><span class="co">## |-- INFO :  Deleting sample.Rmd. </span></span>
<span><span class="co">## |-- INFO :  preparing _bookdown.yml</span></span></code></pre>
<pre><code><span><span class="co">## /usr/local/bin/pandoc +RTS -K512m -RTS _main.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output _main.html --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/latex-div.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/table-classes.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/anchor-sections.lua --metadata-file /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/filed8b910d58acd --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 2 --template /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/bookdown/templates/gitbook.html --highlight-style pygments --css style.css --mathjax --include-in-header /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/rmarkdown-strd8b945887f72.html</span></span></code></pre>
<pre><code><span><span class="co">## |-- INFO :  Results have been copied to: /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/pbmc_report</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="viewing-the-output">Viewing the output<a class="anchor" aria-label="anchor" href="#viewing-the-output"></a>
</h2>
<p>The output will be stored in the out_dir specified. You can view the
report by opening index.html in your browser.</p>
</div>
<div class="section level2">
<h2 id="customization-options">Customization Options<a class="anchor" aria-label="anchor" href="#customization-options"></a>
</h2>
<p>You can tailor the gm_report() using optional parameters:</p>
<ul>
<li>plot_profiles_params, FeaturePlot_params, SpatialDimPlot_params</li>
<li>add_module_score_params to adjust Seuratâ€™s AddModuleScore</li>
<li>Toggle report sections with section argument</li>
<li>Supply an api_key for cell-type guessing via Gemini AI</li>
<li>Use is_spatial_exp=TRUE for spatial transcriptomics data</li>
</ul>
</div>
<div class="section level2">
<h2 id="spatial-experiment">Spatial Experiment<a class="anchor" aria-label="anchor" href="#spatial-experiment"></a>
</h2>
<p>Here we will use a subset of the 10X genomics lymph_node VISIUM (V1)
dataset.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_clusters_2"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- INFO :  Dataset 7870305/files/lymph_node_tiny_clusters_2 has been loaded.</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7870305/files/lymph_node_tiny_2"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- INFO :  Dataset 7870305/files/lymph_node_tiny_2 has been loaded.</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/gm_report.html">gm_report</a></span><span class="op">(</span></span>
<span>  cluster_set <span class="op">=</span> <span class="va">lymph_node_tiny_clusters_2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span>,</span>
<span>  seurat_object <span class="op">=</span> <span class="va">lymph_node_tiny_2</span>,</span>
<span>  smp_species <span class="op">=</span> <span class="st">"Homo sapiens"</span>,</span>
<span>  smp_region <span class="op">=</span> <span class="st">"total"</span>,</span>
<span>  smp_organ <span class="op">=</span> <span class="st">"lymph node"</span>,</span>
<span>  smp_stage <span class="op">=</span> <span class="st">"adult"</span>,</span>
<span>  annotation_src <span class="op">=</span> <span class="st">"CC"</span>,</span>
<span>  bioc_org_db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>  is_spatial_exp <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  SpatialFeaturePlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  SpatialDimPlot_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pt.size.factor <span class="op">=</span> <span class="fl">3000</span><span class="op">)</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_dir</span>, <span class="st">"spatial_report"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## |-- DEBUG :  Checking seurat object. </span></span>
<span><span class="co">## |-- DEBUG :  Checking ClusterSet format. </span></span>
<span><span class="co">## |-- INFO :  No Gemini key provided... </span></span>
<span><span class="co">## |-- INFO :  Canceling IA-based cell type annotation. </span></span>
<span><span class="co">## |-- INFO :  Computing module scores... </span></span>
<span><span class="co">## |-- DEBUG :  Created temporary directory: </span></span>
<span><span class="co">## |-- INFO :  /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-50-34-2025 </span></span>
<span><span class="co">## |-- DEBUG :  Copying file in temporary directory. </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: _bookdown.yml </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: footer.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: index.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: module.Rmd </span></span>
<span><span class="co">## |-- INFO :  The temporary dir contains file: style.css </span></span>
<span><span class="co">## |-- DEBUG :  Computing top genes. </span></span>
<span><span class="co">## |-- DEBUG :  Checking ClusterSet format. </span></span>
<span><span class="co">## |-- DEBUG :  Using fast computation of pearson correlations. </span></span>
<span><span class="co">## |-- DEBUG :  Using fast computation of pearson correlations. </span></span>
<span><span class="co">## |-- INFO :  Results are stored in 'top_genes' slot of the object. </span></span>
<span><span class="co">## |-- DEBUG :  Computing centers. </span></span>
<span><span class="co">## |-- INFO :  Computing centers. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing matrix. </span></span>
<span><span class="co">## |-- DEBUG :  Renaming columns / rows. </span></span>
<span><span class="co">## |-- DEBUG :  Looping over clusters... </span></span>
<span><span class="co">## |-- DEBUG :  Computing cluster 1 center. </span></span>
<span><span class="co">## |-- DEBUG :  Computing cluster 2 center. </span></span>
<span><span class="co">## |-- INFO :  Computing semantic similarity (CC) for cnet_plot. </span></span>
<span><span class="co">## |-- INFO :  Preparing parameters for the report. </span></span>
<span><span class="co">## |-- INFO :  Looping through parameterised Reports. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing Rmd files for modules 1 </span></span>
<span><span class="co">## |-- DEBUG :  file : /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-50-34-2025/rmarkdown/module_0001.Rmd </span></span>
<span><span class="co">## |-- DEBUG :  Preparation of Rmd files for objects 1 finished. </span></span>
<span><span class="co">## |-- DEBUG :  Preparing Rmd files for modules 2 </span></span>
<span><span class="co">## |-- DEBUG :  file : /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-50-34-2025/rmarkdown/module_0002.Rmd </span></span>
<span><span class="co">## |-- DEBUG :  Preparation of Rmd files for objects 2 finished. </span></span>
<span><span class="co">## |-- INFO :  Deleting sample.Rmd. </span></span>
<span><span class="co">## |-- INFO :  preparing _bookdown.yml</span></span></code></pre>
<pre><code><span><span class="co">## Rendering book in directory '/var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrlTue-Jul- 1-18-50-34-2025/rmarkdown'</span></span></code></pre>
<pre><code><span><span class="co">## processing file: _main.Rmd</span></span></code></pre>
<pre><code><span><span class="co">## output file: _main.knit.md</span></span></code></pre>
<pre><code><span><span class="co">## /usr/local/bin/pandoc +RTS -K512m -RTS _main.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output _main.html --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/latex-div.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/table-classes.lua --lua-filter /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/rmarkdown/rmarkdown/lua/anchor-sections.lua --metadata-file /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/filed8b9355cc643 --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 2 --template /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/library/bookdown/templates/gitbook.html --highlight-style pygments --css style.css --mathjax --include-in-header /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/rmarkdown-strd8b916c76ad1.html</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Output created: _book/index.html</span></span></code></pre>
<pre><code><span><span class="co">## |-- INFO :  Results have been copied to: /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpF8EZrl/spatial_report</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The <code><a href="../reference/gm_report.html">gm_report()</a></code> function in Scigenex offers a powerful
way to analyze and document gene modules using Seurat and clusterSet
objects. It supports spatial transcriptomics, functional annotation, and
AI-based interpretation.</p>
<p>For advanced use, explore:</p>
<ul>
<li>?gm_report in R</li>
<li>Scigenex documentation</li>
<li>Customizing Bookdown templates via <code>rmd_dir</code>
argument.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
