<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ClusterSet-class — ClusterSet-class • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ClusterSet-class — ClusterSet-class"><meta property="og:description" content="This class is a representation of a partitioning algorithm and is intented to store gene clusters."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ClusterSet-class</h1>
    <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/R/ClusterSet_class.R"><code>R/ClusterSet_class.R</code></a></small>
    <div class="hidden name"><code>ClusterSet-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This class is a representation of a partitioning algorithm and is intented to store gene clusters.</p>
    </div>


    <div id="value">
    <h2>Value</h2>
    <p>A ClusterSet object.</p>
    </div>
    <div id="slots">
    <h2>Slots</h2>


<dl><dt><code>data</code></dt>
<dd><p>A matrix containing the filtered and partitioned data.</p></dd>


<dt><code>gene_clusters</code></dt>
<dd><p>A list contains the partitioned genes of the dataset. Each element of the list corresponds to a cluster, and contains the indices of the genes assigned to that cluster.</p></dd>


<dt><code>top_genes</code></dt>
<dd><p>A list contains the top genes from the gene clusters. Each element of the list corresponds to a cluster, and contains the indices of the genes assigned to that cluster ranked by their correlation value within their cluster.</p></dd>


<dt><code>gene_clusters_metadata</code></dt>
<dd><p>A list contains metadata related to the gene clusters such as the number of gene clusters, their ID, and the number of genes contained in each of them.</p></dd>


<dt><code>gene_cluster_annotations</code></dt>
<dd><p>A list contains the result obtained from the GO enrichment analysis of gene clusters.</p></dd>


<dt><code>cells_metadata</code></dt>
<dd><p>A list containing metadata related to the cell clusters such as the clustering results the number of cell clusters, their order, colors associated to each cluster,...</p></dd>


<dt><code>dbf_output</code></dt>
<dd><p>A list containing the intermediates outputs of the DBF function : dknn, simulated distances, critical distance and fdr values.</p></dd>


<dt><code>parameters</code></dt>
<dd><p>A list containing the parameter used. Each element of the list correspond to a parameter.</p></dd>


</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SeuratObject</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: sp</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ‘SeuratObject’ was built under R 4.4.0 but the current version is</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4.4.1; it is recomended that you reinstall ‘SeuratObject’ as the ABI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> for R may have changed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘SeuratObject’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     intersect, t</span>
<span class="r-in"><span><span class="fu"><a href="load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Dataset 7871581/files/pbmc3k_medium has been loaded. </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select informative genes</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="select_genes.html">select_genes</a></span><span class="op">(</span><span class="va">pbmc3k_medium</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Number of selected rows/genes (row_sum): 1531 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances using selected method: pearson </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances to KNN. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing simulated distances to KNN. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances to KNN threshold (DKNN threshold). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Selecting informative genes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Instantiating a ClusterSet object. </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Cluster informative features</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="gene_clustering.html">gene_clustering</a></span><span class="op">(</span><span class="va">res</span>, inflation<span class="op">=</span><span class="fl">1.6</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances between selected genes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Compute distances to KNN between selected genes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances to KNN. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Creating the input file for graph partitioning. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Deleting reciprocal edges. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Converting distances into weights. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Writing the input file. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Input file saved in :/var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpGaEOCK/4qPd5kES06.graph_input.txt </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  MCL algorithm has been selected. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  MCL path :  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  mcl_dir: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Number of clusters found: 13 </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/is.html" class="external-link">is</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ClusterSet"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Plot heatmap of gene clusters</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">res</span>, row_labels <span class="op">=</span> <span class="cn">FALSE</span>, line_size_horizontal <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span>, row_labels <span class="op">=</span> <span class="cn">FALSE</span>, line_size_horizontal <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="op">]</span>, row_labels <span class="op">=</span> <span class="cn">FALSE</span>, line_size_horizontal <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, row_labels <span class="op">=</span> <span class="cn">FALSE</span>, line_size_horizontal <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plot the profiles</span></span></span>
<span class="r-in"><span><span class="va">idents</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc3k_medium</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plot_profiles.html">plot_profiles</a></span><span class="op">(</span><span class="va">res</span>,</span></span>
<span class="r-in"><span>              ident <span class="op">=</span> <span class="va">idents</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> |-- STOP : Please run compute_centers() on ClusterSet.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Some methods of the ClusterSet object</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">col_names</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">row_names</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_genes.html">get_genes</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span><span class="op">&lt;-</span>  <span class="fu">clust_size</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IL32"</span>, <span class="st">"CCL5"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">res</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">which_clust</span><span class="op">(</span><span class="va">res</span>, genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IL32"</span>, <span class="st">"CCL5"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">top_genes</span><span class="op">(</span><span class="va">res</span>, top<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  One or several clusters contain less than 5 genes. Retrieving all genes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Results are stored in 'top_genes' slot of the object. </span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">rename_clust</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">clust_names</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "1" "2"</span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span>, <span class="fu">col_names</span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		An object of class ClusterSet</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		Name: 4qPd5kES06 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		Memory used:  2989008 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		Number of cells:  10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		Number of informative genes:  74 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		Number of gene clusters:  2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 		This object contains the following informations:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  data </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  gene_clusters </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  top_genes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  gene_clusters_metadata </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  gene_cluster_annotations </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  cells_metadata </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  dbf_output </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 			 -  parameters </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  distance_method  =  pearson </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  k  =  80 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  noise_level  =  5e-05 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  fdr  =  5e-05 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  row_sum  =  1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  no_dknn_filter  =  FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  seed  =  123 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  keep_nn  =  FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  k_graph  =  5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  output_path  =  /var/folders/zy/wl3dj2_n76zfc8sdvny1q06c0000gn/T//RtmpGaEOCK </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  name  =  4qPd5kES06 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  inflation  =  1.6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 				 *  algorithm  =  mcl </span>
<span class="r-in"><span><span class="fu"><a href="show_methods.html">show_methods</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "%in%"                     "["                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] "centers"                  "clust_names"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] "clust_size"               "cluster_set_to_xls"      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "cluster_stats"            "col_names"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "compute_centers"          "dim"                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "enrich_go"                "gene_cluster"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "grep_clust"               "module_quality_scores"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "nclust"                   "plot_clust_enrichments"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] "plot_ggheatmap"           "plot_markers_to_clusters"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "rename_clust"             "reorder_clust"           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [21] "reorder_genes"            "row_names"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [23] "show"                     "subsample_by_ident"      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "top_by_go"                "top_by_grep"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [27] "top_by_intersect"         "top_genes"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [29] "viz_enrich"               "which_clust"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "write_clust"             </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

