<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Transform a Seurat objects / FindAllMarkers result into a ClusterSet. — cluster_set_from_seurat • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Transform a Seurat objects / FindAllMarkers result into a ClusterSet. — cluster_set_from_seurat"><meta property="og:description" content="Transform a Seurat objects into a ClusterSet."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
<li>
  <a href="../articles/report.html">3 - Creating a bookdown report</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Transform a Seurat objects / FindAllMarkers result into a ClusterSet.</h1>
    <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/R/convert.R"><code>R/convert.R</code></a></small>
    <div class="hidden name"><code>cluster_set_from_seurat.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Transform a Seurat objects into a ClusterSet.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">cluster_set_from_seurat</span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  layer <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"counts"</span>, <span class="st">"scale.data"</span><span class="op">)</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  p_val_adj <span class="op">=</span> <span class="fl">0.001</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>A Seurat object.</p></dd>


<dt id="arg-markers">markers<a class="anchor" aria-label="anchor" href="#arg-markers"></a></dt>
<dd><p>A Seurat::FindAllMarkers() result or a named vector (clusters with gene_names as named).
Named vectors can contains several position with the same gene name (overlapping clusters).</p></dd>


<dt id="arg-layer">layer<a class="anchor" aria-label="anchor" href="#arg-layer"></a></dt>
<dd><p>One of 'data', 'counts' or 'scale.data'. The slot to extract from the seurat object to perform clustering analysis.
SCT is the recommended method from Seurat package when working with spatial transcriptomics data.</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>The type of assay (e.g. "RNA", "Spatial", "Sketch", "SCT"...).</p></dd>


<dt id="arg-p-val-adj">p_val_adj<a class="anchor" aria-label="anchor" href="#arg-p-val-adj"></a></dt>
<dd><p>If markers is the output from Seurat::FindAllMarkers(), the adjusted p-value threshold.</p></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## From a scRNA-seq/Seurat object</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: sp</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ‘SeuratObject’ was built under R 4.4.0 but the current version is</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4.4.1; it is recomended that you reinstall ‘SeuratObject’ as the ABI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> for R may have changed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘SeuratObject’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     intersect, t</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pbmc_small"</span>, package<span class="op">=</span><span class="st">"SeuratObject"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">pbmc_small</span>, only.pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating cluster 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating cluster 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating cluster 2</span>
<span class="r-in"><span><span class="va">cs</span> <span class="op">&lt;-</span> <span class="fu">cluster_set_from_seurat</span><span class="op">(</span><span class="va">pbmc_small</span>, <span class="va">markers</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing centers. </span>
<span class="r-in"><span><span class="va">cs</span> <span class="op">&lt;-</span> <span class="fu">top_genes</span><span class="op">(</span><span class="va">cs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  One or several clusters contain less than 20 genes. Retrieving all genes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Results are stored in 'top_genes' slot of the object. </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">cs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">cs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">cs</span><span class="op">[</span><span class="fl">2</span>, <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  cell_clusters is not NULL. Setting show_dendro to FALSE </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ordering cells/columns using hierarchical clustering. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">cs</span>, cell_clusters <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Extracting cell identity. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plot is interactive... </span>
<span class="r-in"><span><span class="fu"><a href="plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">cs</span><span class="op">[</span><span class="fl">1</span>,<span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span> <span class="op">==</span> <span class="st">"0"</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>cell_clusters <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span>, label_size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>|-- WARNING : The ClusterSet object does not contain any gene.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>|-- WARNING : The ClusterSet object does not contain any gene.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Extracting cell identity. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Centering matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Flooring matrix. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Plotting heatmap. </span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in m[order(nrow(m):1), ]:</span> subscript out of bounds</span>
<span class="r-in"><span><span class="fu"><a href="plot_profiles.html">plot_profiles</a></span><span class="op">(</span><span class="va">cs</span>, ident <span class="op">=</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc_small</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Number of cells types: 3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Number of cells: 80 </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 80 rows containing missing values or values outside the scale range (`geom_col()`).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 1 row containing missing values or values outside the scale range (`geom_text()`).</span>
<span class="r-plt img"><img src="cluster_set_from_seurat-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

