<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
<li>
  <a href="../articles/report.html">3 - Creating a bookdown report</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="ClusterSet-class.html">ClusterSet-class</a></code> </p>
        </td>
        <td><p>ClusterSet-class</p></td>
      </tr><tr><td>
          <p><code><a href="call_infomap_clusterset.html">call_infomap_clusterset()</a></code> </p>
        </td>
        <td><p>Perform Community Detection Using the infomap Algorithm (internal function)</p></td>
      </tr><tr><td>
          <p><code><a href="call_louvain_clusterset.html">call_louvain_clusterset()</a></code> </p>
        </td>
        <td><p>Perform Community Detection Using the Louvain Algorithm (internal function)</p></td>
      </tr><tr><td>
          <p><code><a href="call_walktrap_clusterset.html">call_walktrap_clusterset()</a></code> </p>
        </td>
        <td><p>Perform Community Detection Using the Walktrap Algorithm (internal function)</p></td>
      </tr><tr><td>
          <p><code><a href="centers-ClusterSet-method.html">centers(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Extract centers from a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="check_format_cluster_set.html">check_format_cluster_set()</a></code> </p>
        </td>
        <td><p>Check the format of a Clusterset object</p></td>
      </tr><tr><td>
          <p><code><a href="clust_names-ClusterSet-method.html">clust_names(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>The names of the gene clusters stored in the ClusterSet object</p></td>
      </tr><tr><td>
          <p><code><a href="clust_size-ClusterSet-method.html">clust_size(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Sizes of the clusters stored in a ClusterSet object clust_size</p></td>
      </tr><tr><td>
          <p><code><a href="cluster_set_from_matrix.html">cluster_set_from_matrix()</a></code> </p>
        </td>
        <td><p>Transform any matrix and list into a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="cluster_set_from_seurat.html">cluster_set_from_seurat()</a></code> </p>
        </td>
        <td><p>Transform a Seurat objects / FindAllMarkers result into a ClusterSet.</p></td>
      </tr><tr><td>
          <p><code><a href="cluster_set_report.html">cluster_set_report()</a></code> </p>
        </td>
        <td><p>Create a report from a ClusterSet and Seurat object. DEPRECATED</p></td>
      </tr><tr><td>
          <p><code><a href="cluster_set_to_xls-ClusterSet-method.html">cluster_set_to_xls(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Write Cluster-Set gene lists into an excel sheet.</p></td>
      </tr><tr><td>
          <p><code><a href="cluster_stats-ClusterSet-method.html">cluster_stats(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Compute statistics about the clusters</p></td>
      </tr><tr><td>
          <p><code><a href="cmp_to_a_list.html">cmp_to_a_list()</a></code> </p>
        </td>
        <td><p>Map a user list to a ClusterSet and display various metrics related to overlap with internaly stored clusters</p></td>
      </tr><tr><td>
          <p><code><a href="col_names-ClusterSet-method.html">col_names(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Column names of a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="colors_for_gradient.html">colors_for_gradient()</a></code> </p>
        </td>
        <td><p>Generate a vector of colors for a gradient</p></td>
      </tr><tr><td>
          <p><code><a href="compare_genesets.html">compare_genesets()</a></code> </p>
        </td>
        <td><p>Compare two lists of gene sets</p></td>
      </tr><tr><td>
          <p><code><a href="complex9.html">complex9</a></code> </p>
        </td>
        <td><p>The Complex9 dataset A set of 2D points corresponding to 9 shapes.</p></td>
      </tr><tr><td>
          <p><code><a href="complex9Noisy.html">complex9Noisy</a></code> </p>
        </td>
        <td><p>The complex9Noisy dataset A noisy version of the complex9 dataset. A set of 2D points corresponding to 9 shapes embedded in a noisy environment. A uniform noise was added to the dataset.</p></td>
      </tr><tr><td>
          <p><code><a href="compute_centers-ClusterSet-method.html">compute_centers(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Compute centers of gene modules (e.g mean profiles)</p></td>
      </tr><tr><td>
          <p><code><a href="create_3_rnd_clust.html">create_3_rnd_clust()</a></code> </p>
        </td>
        <td><p>Generate an example dataset with three clusters of profiles</p></td>
      </tr><tr><td>
          <p><code><a href="create_4_rnd_clust.html">create_4_rnd_clust()</a></code> </p>
        </td>
        <td><p>Generate an example dataset with four clusters of profiles</p></td>
      </tr><tr><td>
          <p><code><a href="create_rand_str.html">create_rand_str()</a></code> </p>
        </td>
        <td><p>Generate a random string of letters and numbers</p></td>
      </tr><tr><td>
          <p><code><a href="discrete_palette.html">discrete_palette()</a></code> </p>
        </td>
        <td><p>Generate a discrete color palette</p></td>
      </tr><tr><td>
          <p><code><a href="display_hull.html">display_hull()</a></code> </p>
        </td>
        <td><p>Draw a hull around a region of interest of Visium data.</p></td>
      </tr><tr><td>
          <p><code><a href="do_closest_neighbor_graph.html">do_closest_neighbor_graph()</a></code> </p>
        </td>
        <td><p>Construct a new graph for a ClusterSet object (internal)</p></td>
      </tr><tr><td>
          <p><code><a href="do_reciprocal_neighbor_graph.html">do_reciprocal_neighbor_graph()</a></code> </p>
        </td>
        <td><p>Construct a graph based on selected genes in a ClusterSet object and their neighbors</p></td>
      </tr><tr><td>
          <p><code><a href="enrich_go-ClusterSet-method.html">enrich_go(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Enrichment analysis using GO database of gene clusters from a ClusterSet object</p></td>
      </tr><tr><td>
          <p><code><a href="filter_by_dot_prod.html">filter_by_dot_prod()</a></code> </p>
        </td>
        <td><p>Filter cluster from a ClusterSet object using dot product.</p></td>
      </tr><tr><td>
          <p><code><a href="filter_cluster_sd.html">filter_cluster_sd()</a></code> </p>
        </td>
        <td><p>Filter out gene clusters with small standard deviation.</p></td>
      </tr><tr><td>
          <p><code><a href="filter_cluster_size.html">filter_cluster_size()</a></code> </p>
        </td>
        <td><p>Filter out gene clusters containing few genes</p></td>
      </tr><tr><td>
          <p><code><a href="filter_nb_supporting_cells.html">filter_nb_supporting_cells()</a></code> </p>
        </td>
        <td><p>Filter out cluster supported by few cells.</p></td>
      </tr><tr><td>
          <p><code><a href="gene_cluster-ClusterSet-method.html">gene_cluster(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>The gene clusters stored in a ClusterSet. gene_cluster</p></td>
      </tr><tr><td>
          <p><code><a href="gene_clustering.html">gene_clustering()</a></code> </p>
        </td>
        <td><p>Gene clustering using the Markov Cluster Algorithm (MCL) method.</p></td>
      </tr><tr><td>
          <p><code><a href="getFlippedTissueCoordinates.html">getFlippedTissueCoordinates()</a></code> </p>
        </td>
        <td><p>Seurat object internally store spot coordinates (see Seurat::GetTissueCoordinates()). However, at least in the case of Visium, data are flipped and rotated before SpatialDimPlot. This function return the rotated/flipped tissue Coordinates from a Seurat object.</p></td>
      </tr><tr><td>
          <p><code><a href="get_data_for_scigenex.html">get_data_for_scigenex()</a></code> </p>
        </td>
        <td><p>Fetch an expression matrix from a file, dataframe or Seurat object.</p></td>
      </tr><tr><td>
          <p><code><a href="get_genes.html">get_genes()</a></code> </p>
        </td>
        <td><p>Extract genes list from each gene clusters.</p></td>
      </tr><tr><td>
          <p><code><a href="get_verbosity.html">get_verbosity()</a></code> </p>
        </td>
        <td><p>Get the current verbosity level.</p></td>
      </tr><tr><td>
          <p><code><a href="gm_report.html">gm_report()</a></code> </p>
        </td>
        <td><p>Create a report from a ClusterSet and Seurat objects</p></td>
      </tr><tr><td>
          <p><code><a href="grep_clust-ClusterSet-method.html">grep_clust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Search gene module using a regular expression.</p></td>
      </tr><tr><td>
          <p><code><a href="install_mcl.html">install_mcl()</a></code> </p>
        </td>
        <td><p>Install MCL</p></td>
      </tr><tr><td>
          <p><code><a href="load_example_dataset.html">load_example_dataset()</a></code> </p>
        </td>
        <td><p>Load/download a Seurat or ClusterSet example dataset.</p></td>
      </tr><tr><td>
          <p><code><a href="lymph_node_selected_spot.html">lymph_node_selected_spot</a></code> </p>
        </td>
        <td><p>The lymph_node_selected_spot dataset</p></td>
      </tr><tr><td>
          <p><code><a href="lymph_node_selected_spot_2.html">lymph_node_selected_spot</a></code> </p>
        </td>
        <td><p>The lymph_node_selected_spot dataset (#2)</p></td>
      </tr><tr><td>
          <p><code><a href="mcl_system_cmd.html">mcl_system_cmd()</a></code> </p>
        </td>
        <td><p>Call MCL program for graph partitioning (internal).</p></td>
      </tr><tr><td>
          <p><code><a href="module_quality_scores-ClusterSet-method.html">module_quality_scores(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Compute Gene Module Quality Scores</p></td>
      </tr><tr><td>
          <p><code><a href="nclust-ClusterSet-method.html">nclust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Number of clusters in a ClusterSet object. nclust</p></td>
      </tr><tr><td>
          <p><code><a href="plot_clust_enrichments-ClusterSet-method.html">plot_clust_enrichments(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot Cluster Enrichments</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cluster_stats.html">plot_cluster_stats()</a></code> </p>
        </td>
        <td><p>Plot the result of cluster_stats()</p></td>
      </tr><tr><td>
          <p><code><a href="plot_cmp_genesets.html">plot_cmp_genesets()</a></code> </p>
        </td>
        <td><p>Plot the result of compare_genesets</p></td>
      </tr><tr><td>
          <p><code><a href="plot_dist.html">plot_dist()</a></code> </p>
        </td>
        <td><p>Plot Distribution of distances</p></td>
      </tr><tr><td>
          <p><code><a href="plot_ggheatmap-ClusterSet-method.html">plot_ggheatmap(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Heatmap of data contained in a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_heatmap.html">plot_heatmap()</a></code> </p>
        </td>
        <td><p>plot_heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="plot_markers_to_clusters-ClusterSet-method.html">plot_markers_to_clusters(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Map cell markers to clusters (hypergeometric/jaccard)</p></td>
      </tr><tr><td>
          <p><code><a href="plot_multi_profiles.html">plot_multi_profiles()</a></code> </p>
        </td>
        <td><p>Plot a summarized view of expression profiles.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_profiles.html">plot_profiles()</a></code> </p>
        </td>
        <td><p>Plot mean expression profiles of each cluster from a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_spatial.html">plot_spatial()</a></code> </p>
        </td>
        <td><p>XY scatter plot of Visium data with hexagonal spots.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_spatial_panel.html">plot_spatial_panel()</a></code> </p>
        </td>
        <td><p>A planel of XY scatter plots of Visium data with hexagonal spots.</p></td>
      </tr><tr><td>
          <p><code><a href="plot_stratum.html">plot_stratum()</a></code> </p>
        </td>
        <td><p>Display the results of stratify_seurat</p></td>
      </tr><tr><td>
          <p><code><a href="print_msg.html">print_msg()</a></code> </p>
        </td>
        <td><p>Print a message based on the level of verbosity</p></td>
      </tr><tr><td>
          <p><code><a href="print_stat.html">print_stat()</a></code> </p>
        </td>
        <td><p>Debugging statistics (vector, matrix or dataframe)</p></td>
      </tr><tr><td>
          <p><code><a href="reload_pac.html">reload_pac()</a></code> </p>
        </td>
        <td><p>Reload scigenex (used for development).</p></td>
      </tr><tr><td>
          <p><code><a href="rename_clust-ClusterSet-method.html">rename_clust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Rename the gene clusters of a ClusterSet</p></td>
      </tr><tr><td>
          <p><code><a href="reorder_clust-ClusterSet-method.html">reorder_clust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Reorder the clusters from a ClusterSet</p></td>
      </tr><tr><td>
          <p><code><a href="reorder_genes-ClusterSet-method.html">reorder_genes(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Reorder genes in a ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="row_names-ClusterSet-method.html">row_names(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Row names of a ClusterSet object. row_names</p></td>
      </tr><tr><td>
          <p><code><a href="select_genes.html">select_genes()</a></code> </p>
        </td>
        <td><p>Selects informative genes based on k-nearest neighbour analysis.</p></td>
      </tr><tr><td>
          <p><code><a href="set_verbosity.html">set_verbosity()</a></code> </p>
        </td>
        <td><p>Set the verbosity level for the SciGeneX package</p></td>
      </tr><tr><td>
          <p><code><a href="show_methods.html">show_methods()</a></code> </p>
        </td>
        <td><p>List the method for the ClusterSet object.</p></td>
      </tr><tr><td>
          <p><code><a href="stratify_seurat.html">stratify_seurat()</a></code> </p>
        </td>
        <td><p>Given two spot groups, compute for each spot: (i) the distance of the closest spot in the other class.</p></td>
      </tr><tr><td>
          <p><code><a href="subsample_by_ident-ClusterSet-method.html">subsample_by_ident(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Given ncell, a target number, select ncell from each class of cell/column.</p></td>
      </tr><tr><td>
          <p><code><a href="top_by_go-ClusterSet-method.html">top_by_go(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Select top_genes based on go terms</p></td>
      </tr><tr><td>
          <p><code><a href="top_by_grep-ClusterSet-method.html">top_by_grep(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Select top_genes based on a regular expression search</p></td>
      </tr><tr><td>
          <p><code><a href="top_by_intersect-ClusterSet-method.html">top_by_intersect(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Select top_genes based on intersection with a list.</p></td>
      </tr><tr><td>
          <p><code><a href="top_genes-ClusterSet-method.html">top_genes(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Most co-expressed genes from each gene cluster</p></td>
      </tr><tr><td>
          <p><code><a href="viz_enrich-ClusterSet-method.html">viz_enrich(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Visualization of enrichment analyses results from a ClusterSet object</p></td>
      </tr><tr><td>
          <p><code><a href="which_clust-ClusterSet-method.html">which_clust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Which clusters contain a set of genes.</p></td>
      </tr><tr><td>
          <p><code><a href="write_clust-ClusterSet-method.html">write_clust(<i>&lt;ClusterSet&gt;</i>)</a></code> </p>
        </td>
        <td><p>Write the cluster to files.</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

