<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Heatmap of data contained in a ClusterSet object. — plot_ggheatmap • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Heatmap of data contained in a ClusterSet object. — plot_ggheatmap"><meta property="og:description" content="Plot the results (heatmap) contained in a ClusterSet object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Heatmap of data contained in a ClusterSet object.</h1>
    <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/R/visualization.R"><code>R/visualization.R</code></a></small>
    <div class="hidden name"><code>plot_ggheatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot the results (heatmap) contained in a ClusterSet object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_ggheatmap</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  to_log2 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_top_genes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ident <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  panel_spacing <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  colors <span class="op">=</span> <span class="fu"><a href="colors_for_gradient.html">colors_for_gradient</a></span><span class="op">(</span><span class="st">"Ju1"</span><span class="op">)</span>,</span>
<span>  standardizing <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  color_ident <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ceil <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  floor <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>  centering <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Genes"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Spots"</span>,</span>
<span>  hide_gene_name <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  hide_col_name <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pseudocount <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  coord_flip <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>A ClusterSet object.</p></dd>


<dt id="arg-to-log-">to_log2<a class="anchor" aria-label="anchor" href="#arg-to-log-"></a></dt>
<dd><p>Whether data should be transform in logarithm base 2 (+ 0.1 as a pseudocount).</p></dd>


<dt id="arg-use-top-genes">use_top_genes<a class="anchor" aria-label="anchor" href="#arg-use-top-genes"></a></dt>
<dd><p>A logical to indicate whether to use highly similar genes in the slot top_genes of ClusterSet.</p></dd>


<dt id="arg-ident">ident<a class="anchor" aria-label="anchor" href="#arg-ident"></a></dt>
<dd><p>A named vector containing the cell type identities for each cell.
Typically the result from the Idents() function on a Seurat object (see Seurat library).</p></dd>


<dt id="arg-panel-spacing">panel_spacing<a class="anchor" aria-label="anchor" href="#arg-panel-spacing"></a></dt>
<dd><p>Spacing between facets/panels ("line" units).</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>A vector of colors for the gradient.</p></dd>


<dt id="arg-standardizing">standardizing<a class="anchor" aria-label="anchor" href="#arg-standardizing"></a></dt>
<dd><p>Whether rows should be divided by standard deviation.</p></dd>


<dt id="arg-color-ident">color_ident<a class="anchor" aria-label="anchor" href="#arg-color-ident"></a></dt>
<dd><p>A vector containing colors for the cell classes as.</p></dd>


<dt id="arg-ceil">ceil<a class="anchor" aria-label="anchor" href="#arg-ceil"></a></dt>
<dd><p>A value for ceiling (NULL for no ceiling). Ceiling is performed after log transformation, centering and standardization.</p></dd>


<dt id="arg-floor">floor<a class="anchor" aria-label="anchor" href="#arg-floor"></a></dt>
<dd><p>A value for flooring (NULL for no flooring). Flooring is performed after log transformation, centering and standardization.</p></dd>


<dt id="arg-centering">centering<a class="anchor" aria-label="anchor" href="#arg-centering"></a></dt>
<dd><p>Whether rows should be centered.</p></dd>


<dt id="arg-xlab">xlab<a class="anchor" aria-label="anchor" href="#arg-xlab"></a></dt>
<dd><p>A name for the x axis.</p></dd>


<dt id="arg-ylab">ylab<a class="anchor" aria-label="anchor" href="#arg-ylab"></a></dt>
<dd><p>A name for the y axis.</p></dd>


<dt id="arg-hide-gene-name">hide_gene_name<a class="anchor" aria-label="anchor" href="#arg-hide-gene-name"></a></dt>
<dd><p>Whether to hide gene names.</p></dd>


<dt id="arg-hide-col-name">hide_col_name<a class="anchor" aria-label="anchor" href="#arg-hide-col-name"></a></dt>
<dd><p>Whether to hide column names.</p></dd>


<dt id="arg-pseudocount">pseudocount<a class="anchor" aria-label="anchor" href="#arg-pseudocount"></a></dt>
<dd><p>A value for the pseudocount added before log transformation.</p></dd>


<dt id="arg-coord-flip">coord_flip<a class="anchor" aria-label="anchor" href="#arg-coord-flip"></a></dt>
<dd><p>Whether to flip the coordinates.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A ggplot diagram.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Load datasets</span></span></span>
<span class="r-in"><span><span class="fu"><a href="load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">'7871581/files/pbmc3k_medium_clusters'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Dataset 7871581/files/pbmc3k_medium_clusters was already loaded. </span>
<span class="r-in"><span><span class="fu"><a href="load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">'7871581/files/pbmc3k_medium'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Dataset 7871581/files/pbmc3k_medium was already loaded. </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># rename clusters</span></span></span>
<span class="r-in"><span><span class="va">new_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="rename_clust.html">rename_clust</a></span><span class="op">(</span><span class="va">pbmc3k_medium_clusters</span>, new_name<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"M%02d"</span>, <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="clust_names.html">clust_names</a></span><span class="op">(</span><span class="va">pbmc3k_medium_clusters</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use plot_ggheatmap</span></span></span>
<span class="r-in"><span><span class="va">ident_pbmc3k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">pbmc3k_medium</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">new_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="top_genes.html">top_genes</a></span><span class="op">(</span><span class="va">new_obj</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  One or several clusters contain less than 20 genes. Retrieving all genes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Results are stored in 'top_genes' slot of the object. </span>
<span class="r-in"><span><span class="fu">plot_ggheatmap</span><span class="op">(</span><span class="va">new_obj</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ident_pbmc3k</span><span class="op">)</span><span class="op">]</span>, ident<span class="op">=</span><span class="va">ident_pbmc3k</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plot_ggheatmap-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

