<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot the result of compare_genesets — plot_cmp_genesets • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the result of compare_genesets — plot_cmp_genesets"><meta property="og:description" content="This function plots the results of compare_genesets."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the result of compare_genesets</h1>
    <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/R/compare_gene_sets.R"><code>R/compare_gene_sets.R</code></a></small>
    <div class="hidden name"><code>plot_cmp_genesets.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function plots the results of compare_genesets.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_cmp_genesets</span><span class="op">(</span></span>
<span>  set_1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  set_2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"jaccard"</span>, <span class="st">"hypergeom"</span>, <span class="st">"intersection"</span>, <span class="st">"union"</span>, <span class="st">"size_set_1"</span>, <span class="st">"size_set_2"</span>,</span>
<span>    <span class="st">"diff_set_1"</span>, <span class="st">"diff_set_2"</span><span class="op">)</span>,</span>
<span>  transform <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"None"</span>, <span class="st">"log10"</span>, <span class="st">"log2"</span>, <span class="st">"-log10"</span>, <span class="st">"-log2"</span><span class="op">)</span>,</span>
<span>  colors <span class="op">=</span> <span class="fu"><a href="colors_for_gradient.html">colors_for_gradient</a></span><span class="op">(</span><span class="st">"Ju1"</span><span class="op">)</span>,</span>
<span>  layout <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"raster"</span>, <span class="st">"square"</span><span class="op">)</span>,</span>
<span>  background <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  coord_equal <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-set-">set_1<a class="anchor" aria-label="anchor" href="#arg-set-"></a></dt>
<dd><p>A list containing gene sets to be compared.</p></dd>


<dt id="arg-set-">set_2<a class="anchor" aria-label="anchor" href="#arg-set-"></a></dt>
<dd><p>A list containing gene sets to be compared.</p></dd>


<dt id="arg-stat">stat<a class="anchor" aria-label="anchor" href="#arg-stat"></a></dt>
<dd><p>The statistics to be computed between gene sets. It can be either "jaccard", "hypergeom", "intersection"
"size_set_1", "size_set_2", "diff_set_1" (specific to set_1), "diff_set_2" (specific to set_2). The background
is taken into account. Note that hypergeometric tests check for enrichment.</p></dd>


<dt id="arg-transform">transform<a class="anchor" aria-label="anchor" href="#arg-transform"></a></dt>
<dd><p>The transformation to be applied to the values. It can be either "NA", "log10", or "log2", "-log10", "-log2.</p></dd>


<dt id="arg-colors">colors<a class="anchor" aria-label="anchor" href="#arg-colors"></a></dt>
<dd><p>The color palette to be used in the plot.</p></dd>


<dt id="arg-layout">layout<a class="anchor" aria-label="anchor" href="#arg-layout"></a></dt>
<dd><p>The type of diagram. Either "raster" (a scatter plot showing the statistics of interest) or "square".
The "square" layout shows the hypergeometric pvalue (color) and the Jaccard result (size of the square).</p></dd>


<dt id="arg-background">background<a class="anchor" aria-label="anchor" href="#arg-background"></a></dt>
<dd><p>The background (universe) to consider. Default to the non-redundant list of elements
merged from set_1 and set2. You may provide a vector with all genes of the genome for instance.</p></dd>


<dt id="arg-coord-equal">coord_equal<a class="anchor" aria-label="anchor" href="#arg-coord-equal"></a></dt>
<dd><p>make sure that the an equal length on both axis represents the same change in units</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A ggplot object representing the comparison results.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>see compare_genesets.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">set_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">letters</span><span class="op">[</span><span class="fl">11</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, <span class="va">letters</span><span class="op">[</span><span class="fl">21</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">set_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, <span class="va">letters</span><span class="op">[</span><span class="fl">21</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="compare_genesets.html">compare_genesets</a></span><span class="op">(</span><span class="va">set_1</span>, <span class="va">set_2</span>, stat <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing background from the union of set_1 and set_2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Background size : 27 </span>
<span class="r-in"><span><span class="fu">plot_cmp_genesets</span><span class="op">(</span><span class="va">set_1</span>, <span class="va">set_2</span>, stat <span class="op">=</span> <span class="st">"jaccard"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Using transformation: None </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing background from the union of set_1 and set_2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Background size : 27 </span>
<span class="r-plt img"><img src="plot_cmp_genesets-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">plot_cmp_genesets</span><span class="op">(</span><span class="va">set_1</span>, <span class="va">set_2</span>, stat <span class="op">=</span> <span class="st">"hypergeom"</span>, transform <span class="op">=</span> <span class="st">"log10"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Using transformation: log10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing background from the union of set_1 and set_2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Background size : 27 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling pvalue to 1e-320 (R limit). </span>
<span class="r-plt img"><img src="plot_cmp_genesets-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">plot_cmp_genesets</span><span class="op">(</span><span class="va">set_1</span>, <span class="va">set_2</span>, layout<span class="op">=</span><span class="st">"square"</span>, transform <span class="op">=</span> <span class="st">"-log10"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Using transformation: -log10 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing background from the union of set_1 and set_2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Background size : 27 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Ceiling pvalue to 1e-320 (R limit). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing background from the union of set_1 and set_2. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Background size : 27 </span>
<span class="r-plt img"><img src="plot_cmp_genesets-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

