<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Selects informative genes based on k-nearest neighbour analysis. — select_genes • scigenex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Selects informative genes based on k-nearest neighbour analysis. — select_genes"><meta property="og:description" content="This function selects genes based on k-nearest neighbour analysis.
The function takes a seurat object or gene expression matrix as input
and compute distance to k-nearest neighbour for each gene/feature.
A threshold is set based on permutation analysis and FDR computation."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scigenex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home"></span>

  </a>
</li>
<li>
  <a href="../articles/usage.html">1 - scRNA-seq guided tutorial</a>
</li>
<li>
  <a href="../articles/spatial.html">2 - Spatial transcriptomics guided tutorial</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dputhier/scigenex/">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Selects informative genes based on k-nearest neighbour analysis.</h1>
    <small class="dont-index">Source: <a href="https://github.com/dputhier/scigenex/blob/HEAD/R/select_genes.R"><code>R/select_genes.R</code></a></small>
    <div class="hidden name"><code>select_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function selects genes based on k-nearest neighbour analysis.
The function takes a seurat object or gene expression matrix as input
and compute distance to k-nearest neighbour for each gene/feature.
A threshold is set based on permutation analysis and FDR computation.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">select_genes</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  distance_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pearson"</span>, <span class="st">"cosine"</span>, <span class="st">"euclidean"</span>, <span class="st">"spearman"</span>, <span class="st">"kendall"</span><span class="op">)</span>,</span>
<span>  noise_level <span class="op">=</span> <span class="fl">5e-05</span>,</span>
<span>  k <span class="op">=</span> <span class="fl">80</span>,</span>
<span>  row_sum <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  fdr <span class="op">=</span> <span class="fl">0.005</span>,</span>
<span>  which_slot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"sct"</span>, <span class="st">"counts"</span><span class="op">)</span>,</span>
<span>  no_dknn_filter <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  no_anti_cor <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">123</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>A matrix, data.frame or Seurat object.</p></dd>


<dt id="arg-distance-method">distance_method<a class="anchor" aria-label="anchor" href="#arg-distance-method"></a></dt>
<dd><p>a character string indicating the method for computing distances (one of "pearson", "cosine",
"euclidean", spearman or "kendall").</p></dd>


<dt id="arg-noise-level">noise_level<a class="anchor" aria-label="anchor" href="#arg-noise-level"></a></dt>
<dd><p>This parameter controls the fraction of genes with high dknn (ie. noise) whose neighborhood (i.e associated distances)
will be used to compute simulated DKNN values. A value of 0 means to use all the genes. A value close to 1 means  to use only gene
with high dknn (i.e close to noise).</p></dd>


<dt id="arg-k">k<a class="anchor" aria-label="anchor" href="#arg-k"></a></dt>
<dd><p>An integer specifying the size of the neighborhood.</p></dd>


<dt id="arg-row-sum">row_sum<a class="anchor" aria-label="anchor" href="#arg-row-sum"></a></dt>
<dd><p>A feature/gene whose row sum is below this threshold will be discarded. Use -Inf to keep all genes.</p></dd>


<dt id="arg-fdr">fdr<a class="anchor" aria-label="anchor" href="#arg-fdr"></a></dt>
<dd><p>A numeric value indicating the false discovery rate threshold (range: 0 to 100).</p></dd>


<dt id="arg-which-slot">which_slot<a class="anchor" aria-label="anchor" href="#arg-which-slot"></a></dt>
<dd><p>a character string indicating which slot to use from the input scRNA-seq object (one of "data", "sct" or "counts").</p></dd>


<dt id="arg-no-dknn-filter">no_dknn_filter<a class="anchor" aria-label="anchor" href="#arg-no-dknn-filter"></a></dt>
<dd><p>a logical indicating whether to skip the k-nearest-neighbors (KNN) filter. If FALSE, all genes are kept for the next steps.</p></dd>


<dt id="arg-no-anti-cor">no_anti_cor<a class="anchor" aria-label="anchor" href="#arg-no-anti-cor"></a></dt>
<dd><p>If TRUE, correlation below 0 are set to zero ("pearson", "cosine", "spearman" "kendall"). This may increase the
relative weight of positive correlation (as true anti-correlation may be rare).</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>An integer specifying the random seed to use.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a ClusterSet class object</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>- Lopez F.,Textoris J., Bergon A., Didier G., Remy E., Granjeaud
S., Imbert J. , Nguyen C. and Puthier D. TranscriptomeBrowser: a powerful
and flexible toolbox to explore productively the transcriptional landscape
of the Gene Expression Omnibus database. PLoSONE, 2008;3(12):e4001.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Julie Bavais, Sebastien Nin, Lionel Spinelli and Denis Puthier</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Restrict vebosity to info messages only.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="set_verbosity.html">set_verbosity</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load a dataset</span></span></span>
<span class="r-in"><span><span class="fu"><a href="load_example_dataset.html">load_example_dataset</a></span><span class="op">(</span><span class="st">"7871581/files/pbmc3k_medium"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Dataset 7871581/files/pbmc3k_medium was already loaded. </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select informative genes</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">select_genes</span><span class="op">(</span><span class="va">pbmc3k_medium</span>,</span></span>
<span class="r-in"><span>                    distance <span class="op">=</span> <span class="st">"pearson"</span>,</span></span>
<span class="r-in"><span>                    row_sum<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances using selected method: pearson </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing distances to KNN. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing simulated distances to KNN. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Computing threshold of distances to KNN (DKNN threshold). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Selecting informative genes. </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |-- INFO :  Creating the ClusterSet object. </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Result is a ClusterSet object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/is.html" class="external-link">is</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ClusterSet"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/methods/slot.html" class="external-link">slotNames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "data"                     "gene_clusters"           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3] "top_genes"                "gene_clusters_metadata"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "gene_cluster_annotations" "cells_metadata"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [7] "dbf_output"               "parameters"              </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># The selected genes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 293</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="row_names.html">row_names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "PTCRA"     "ACRBP"     "TUBB1"     "SDPR"      "HIST1H2AC" "C2orf88"  </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by J. Bavais.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

